{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0bff45d6-9132-49be-8084-775cc63ee625",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1a525e8-72be-46bb-b10e-dc6128c5bb52",
   "metadata": {},
   "source": [
    "## Create subdirectories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "30614ea3-3fe3-4dc8-a846-362196fc4abb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mkdir sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "988179d4-d99e-4844-950e-57f0a75ae517",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mkdir reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9dfa2745-16a3-4e51-9630-d4af14c130f2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mkdir results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6eb48218-7883-4243-ad7a-d0beb9836eee",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mNGS_DNAseq_analysis.ipynb\u001b[0m*  \u001b[01;32mUntitled.ipynb\u001b[0m*  \u001b[34;42mreference\u001b[0m/  \u001b[34;42mresults\u001b[0m/  \u001b[34;42msequences\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff276be2-8879-48b7-b987-9483511889f6",
   "metadata": {},
   "source": [
    "## Go to results folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "822444b4-4235-4d3a-ae2d-1e4bed47ded1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results\n"
     ]
    }
   ],
   "source": [
    "cd results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d878efd9-a5a5-4f7e-b23f-d6c5d563a3c3",
   "metadata": {},
   "source": [
    "## Create more subdirectories in the results folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ea6e012f-aed8-4599-b22f-3925c8353b95",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mkdir fastqc sam bam vcf bcf trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "48b63441-eaa8-4c26-af3e-fac730ff7a4d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[34;42mbam\u001b[0m/  \u001b[34;42mbcf\u001b[0m/  \u001b[34;42mfastqc\u001b[0m/  \u001b[34;42msam\u001b[0m/  \u001b[34;42mtrimmomatic\u001b[0m/  \u001b[34;42mvcf\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c0b73cce-f312-43c4-8b59-60253502d785",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64206636-789b-4f3a-9f33-eb3b574a61f9",
   "metadata": {},
   "source": [
    "## Moving into sequences folder and download fastq files with curl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "be641b14-947c-4fcf-b800-364759b1e67a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/sequences\n"
     ]
    }
   ],
   "source": [
    "cd sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6655e323-54fd-4d72-a3f1-42b366f35820",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  123M  100  123M    0     0  13.7M      0  0:00:08  0:00:08 --:--:-- 17.9M\n"
     ]
    }
   ],
   "source": [
    "!curl -O ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/004/SRR2589044/SRR2589044_1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "da9eca26-a8aa-4565-ace0-2ac17e9ccdf5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  127M  100  127M    0     0  11.8M      0  0:00:10  0:00:10 --:--:-- 14.8M\n"
     ]
    }
   ],
   "source": [
    "!curl -O ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/004/SRR2589044/SRR2589044_2.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52156024-c075-41ad-b6cd-0ae41dbe6f66",
   "metadata": {},
   "source": [
    "## Unzipping downloaded files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "eff730ed-8ebb-47cc-88b8-3222919e9dff",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/sequences'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d7dbaeed-c31e-4a4c-85b0-6eb0fcd6350b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044_1.fastq.gz\u001b[0m*  \u001b[01;32mSRR2589044_2.fastq.gz\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f551ea1-bac9-46e5-b7ef-99a07d07bd00",
   "metadata": {},
   "source": [
    "## unzip any file with .fastq.gz extention"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9613c065-567e-4ec6-bdd4-a206bc4254ef",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!gunzip *.fastq.gz  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b6a50f6e-54e5-4ed3-9d66-00469de0cf6f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044_1.fastq\u001b[0m*  \u001b[01;32mSRR2589044_2.fastq\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a11b197-2f19-4b12-8378-28d96af0eee9",
   "metadata": {},
   "source": [
    "## check first/last 10 lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "22544991-d2f2-472d-8cf8-e86e7ba9164a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SRR2589044.1 HWI-ST957:244:H73TDADXX:1:1101:10469:2228/1\n",
      "GTGGAAACCAGCGACGGTGACGGCTATATCAACTGCGTGATTGAACATCAAAGCTCTGCAGAAAAGAATATGGCTTTTCGGCTAATGCGCTATGCCACTGCCGCCATGCAGCGTCACCTGGATAACTGTCTCTTATACACATCTCCGAGC\n",
      "+\n",
      "BBBFFFFFHHHHHJJJJIIGIIJJJJJGIJJIJJJIJIJGHIIJIIFFFGHHFFFFFEEEEEEDDDDDDDEEDCDDDDDD@BDDDDDDDDBDDDDDDDDDCABDDDDCDDCCDBDDDDDDDBDEDDDCCDCDCDDDDCDDDCDDDD@DDD\n",
      "@SRR2589044.2 HWI-ST957:244:H73TDADXX:1:1101:10877:2125/1\n",
      "GCAATATTACCGGGCGACATCGCCGAAGATGACATCACAGCGGTACTCTGTTTTGTGATTGAAGCCGATCAGATTACCTTTGAAACAGTAGAAGTCTCGCCAAAAATATCCACCCCACCAGTGCTTAAACTGGCAGCCGAACAAGCGCCA\n",
      "+\n",
      "@@CFDDFFDB>DHGEBEADGID;GEAFEFEHGEGCHGEGECHF'9=>CCCCCCCA5=ADCC@CCCBC>8B>CCCCCCCCCCCCCCCCC@@C>AACCC@?@B<BCBBACEDECCCBCBB@@B>CCECCCCAACCBCBBBB>9>BCA8<9<B\n",
      "@SRR2589044.3 HWI-ST957:244:H73TDADXX:1:1101:16059:2143/1\n",
      "TCCCGGTGGCGCGCCACAGTGCGCGGGCCCCCGCCATATTACGGCCATGGGTGGTGGTGGCGGAACGGTACTTAGGCATACTTTATTTACTCCCAGTGTCTGTCTCGTAAATGGGACGGTGCGTGCCGTCCCATTTTTTGTATTTATTGA\n"
     ]
    }
   ],
   "source": [
    "!head SRR2589044_1.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3b51a76d-7cf8-42ed-a5b2-6e5fd9d92665",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+\n",
      "CCCFFFFFHHHHHHIJJJJIJJJJJJJJJJJJJGIJJJJJJJJJHIJJIJJJJHHHHHFCFFFEDEDDDDDDDDDDDDDDDDDD59ACC>>>@@AB+2:5555>>C>CCD>:>ACAA>>&8>8@CA:A9@BBD#################\n",
      "@SRR2589044.1107089 HWI-ST957:245:H73R4ADXX:2:2216:19883:100785/1\n",
      "GGTATGGCTGGAACAACGGTAGCGGCATTGGGCTTTGCCCCGAAGCAAGCACTGGCTCAGGCGCGAAACTACAAATTATTACGCGCTAAAGAGATCCGTAACACCTGCACATACTGTTCCGTAGGTTGCGGGCCATTGATGTATAGCCTT\n",
      "+\n",
      "@@@DDDDDDFF?F;ABB:GBE)?CGA?:D<FGF>BF<@FFI/;A':AEB=BB;;@C9A>>A?=@?8@B-4@A4>A4@B@BB@8999)09@(43934><??<8:AB338+:>::::::A(@@BB08:A?B>>><&8>@AA###########\n",
      "@SRR2589044.1107090 HWI-ST957:245:H73R4ADXX:2:2216:20775:100825/1\n",
      "GTCTGGAAAGGCGCGCGATACAGGGTGACAGCCCCGTACACAAAAATGCACATATTGTGAGCTCGATGAGTAGGGCGGGACACGTGGTATCCTGTCTGAATATGGGGGGACCATCCTCCAAGGCTAAATACTCCTGACTGACCGATAGTG\n",
      "+\n",
      "BCBFFFFFHHHHHJJIIIDIIJJJJFHJJJIIJJJJJIJJJHHHHFFFFFEEEEEEEEDDDDDDBDDEDDDDDDDCDDDDDDDDABBBDDDDEDDDDDDDDEDEDDDDDBDDDDDDDDDDDD@BDDCDCECDDDDDDDDDDDDB@B9@DE\n"
     ]
    }
   ],
   "source": [
    "!tail SRR2589044_1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ca7c411-c4c7-4075-aada-e13d947fa5e4",
   "metadata": {},
   "source": [
    "## count the number of lines in the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "20716bed-983e-4cd1-a439-177b2315630a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4428360\n"
     ]
    }
   ],
   "source": [
    "!cat SRR2589044_1.fastq | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f00cb334-53fa-4425-813c-8d6bb28b2eef",
   "metadata": {},
   "source": [
    "## install fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1cf41c0b-a5fc-4a20-94f5-fa76ca9873a6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.7.4\n",
      "  latest version: 23.9.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.9.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/sdogan/anaconda3/envs/NGS_DNAseq_analysis\n",
      "\n",
      "  added / updated specs:\n",
      "    - fastqc\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    alsa-lib-1.2.10            |       hd590300_0         542 KB  conda-forge\n",
      "    cairo-1.18.0               |       h3faef2a_0         959 KB  conda-forge\n",
      "    expat-2.5.0                |       hcb278e6_1         134 KB  conda-forge\n",
      "    fastqc-0.12.1              |       hdfd78af_0        11.1 MB  bioconda\n",
      "    font-ttf-dejavu-sans-mono-2.37|       hab24e00_0         388 KB  conda-forge\n",
      "    font-ttf-inconsolata-3.000 |       h77eed37_0          94 KB  conda-forge\n",
      "    font-ttf-source-code-pro-2.038|       h77eed37_0         684 KB  conda-forge\n",
      "    font-ttf-ubuntu-0.83       |       hab24e00_0         1.9 MB  conda-forge\n",
      "    fontconfig-2.14.2          |       h14ed4e7_0         266 KB  conda-forge\n",
      "    fonts-conda-ecosystem-1    |                0           4 KB  conda-forge\n",
      "    fonts-conda-forge-1        |                0           4 KB  conda-forge\n",
      "    freetype-2.12.1            |       h267a509_2         620 KB  conda-forge\n",
      "    gettext-0.21.1             |       h27087fc_0         4.1 MB  conda-forge\n",
      "    giflib-5.2.1               |       h0b41bf4_3          76 KB  conda-forge\n",
      "    graphite2-1.3.13           |    h58526e2_1001         102 KB  conda-forge\n",
      "    harfbuzz-8.2.1             |       h3d44ed6_0         1.5 MB  conda-forge\n",
      "    icu-73.2                   |       h59595ed_0        11.5 MB  conda-forge\n",
      "    keyutils-1.6.1             |       h166bdaf_0         115 KB  conda-forge\n",
      "    krb5-1.21.2                |       h659d440_0         1.3 MB  conda-forge\n",
      "    lcms2-2.15                 |       hb7c19ff_3         234 KB  conda-forge\n",
      "    lerc-4.0.0                 |       h27087fc_0         275 KB  conda-forge\n",
      "    libcups-2.3.3              |       h4637d8d_4         4.3 MB  conda-forge\n",
      "    libdeflate-1.19            |       hd590300_0          66 KB  conda-forge\n",
      "    libedit-3.1.20191231       |       he28a2e2_2         121 KB  conda-forge\n",
      "    libglib-2.78.0             |       hebfc3b9_0         2.6 MB  conda-forge\n",
      "    libiconv-1.17              |       h166bdaf_0         1.4 MB  conda-forge\n",
      "    libjpeg-turbo-3.0.0        |       hd590300_1         604 KB  conda-forge\n",
      "    libpng-1.6.39              |       h753d276_0         276 KB  conda-forge\n",
      "    libtiff-4.6.0              |       ha9c0a0a_2         277 KB  conda-forge\n",
      "    libwebp-base-1.3.2         |       hd590300_0         392 KB  conda-forge\n",
      "    libxcb-1.15                |       h0b41bf4_0         375 KB  conda-forge\n",
      "    openjdk-21.0.0             |       haa376d0_0       172.9 MB  conda-forge\n",
      "    pcre2-10.40                |       hc3806b6_0         2.3 MB  conda-forge\n",
      "    perl-5.32.1                | 4_hd590300_perl5        12.7 MB  conda-forge\n",
      "    pixman-0.42.2              |       h59595ed_0         376 KB  conda-forge\n",
      "    pthread-stubs-0.4          |    h36c2ea0_1001           5 KB  conda-forge\n",
      "    xorg-fixesproto-5.0        |    h7f98852_1002           9 KB  conda-forge\n",
      "    xorg-inputproto-2.3.2      |    h7f98852_1002          19 KB  conda-forge\n",
      "    xorg-kbproto-1.0.7         |    h7f98852_1002          27 KB  conda-forge\n",
      "    xorg-libice-1.1.1          |       hd590300_0          57 KB  conda-forge\n",
      "    xorg-libsm-1.2.4           |       h7391055_0          27 KB  conda-forge\n",
      "    xorg-libx11-1.8.7          |       h8ee46fc_0         809 KB  conda-forge\n",
      "    xorg-libxau-1.0.11         |       hd590300_0          14 KB  conda-forge\n",
      "    xorg-libxdmcp-1.1.3        |       h7f98852_0          19 KB  conda-forge\n",
      "    xorg-libxext-1.3.4         |       h0b41bf4_2          49 KB  conda-forge\n",
      "    xorg-libxfixes-5.0.3       |    h7f98852_1004          18 KB  conda-forge\n",
      "    xorg-libxi-1.7.10          |       h7f98852_0          46 KB  conda-forge\n",
      "    xorg-libxrender-0.9.11     |       hd590300_0          37 KB  conda-forge\n",
      "    xorg-libxt-1.3.0           |       hd590300_1         370 KB  conda-forge\n",
      "    xorg-libxtst-1.2.3         |    h7f98852_1002          31 KB  conda-forge\n",
      "    xorg-recordproto-1.14.2    |    h7f98852_1002           8 KB  conda-forge\n",
      "    xorg-renderproto-0.11.1    |    h7f98852_1002           9 KB  conda-forge\n",
      "    xorg-xextproto-7.3.0       |    h0b41bf4_1003          30 KB  conda-forge\n",
      "    xorg-xproto-7.0.31         |    h7f98852_1007          73 KB  conda-forge\n",
      "    zlib-1.2.13                |       hd590300_5          91 KB  conda-forge\n",
      "    zstd-1.5.5                 |       hfc55251_0         532 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       236.6 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  alsa-lib           conda-forge/linux-64::alsa-lib-1.2.10-hd590300_0 \n",
      "  cairo              conda-forge/linux-64::cairo-1.18.0-h3faef2a_0 \n",
      "  expat              conda-forge/linux-64::expat-2.5.0-hcb278e6_1 \n",
      "  fastqc             bioconda/noarch::fastqc-0.12.1-hdfd78af_0 \n",
      "  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0 \n",
      "  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0 \n",
      "  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0 \n",
      "  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-hab24e00_0 \n",
      "  fontconfig         conda-forge/linux-64::fontconfig-2.14.2-h14ed4e7_0 \n",
      "  fonts-conda-ecosy~ conda-forge/noarch::fonts-conda-ecosystem-1-0 \n",
      "  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-0 \n",
      "  freetype           conda-forge/linux-64::freetype-2.12.1-h267a509_2 \n",
      "  gettext            conda-forge/linux-64::gettext-0.21.1-h27087fc_0 \n",
      "  giflib             conda-forge/linux-64::giflib-5.2.1-h0b41bf4_3 \n",
      "  graphite2          conda-forge/linux-64::graphite2-1.3.13-h58526e2_1001 \n",
      "  harfbuzz           conda-forge/linux-64::harfbuzz-8.2.1-h3d44ed6_0 \n",
      "  icu                conda-forge/linux-64::icu-73.2-h59595ed_0 \n",
      "  keyutils           conda-forge/linux-64::keyutils-1.6.1-h166bdaf_0 \n",
      "  krb5               conda-forge/linux-64::krb5-1.21.2-h659d440_0 \n",
      "  lcms2              conda-forge/linux-64::lcms2-2.15-hb7c19ff_3 \n",
      "  lerc               conda-forge/linux-64::lerc-4.0.0-h27087fc_0 \n",
      "  libcups            conda-forge/linux-64::libcups-2.3.3-h4637d8d_4 \n",
      "  libdeflate         conda-forge/linux-64::libdeflate-1.19-hd590300_0 \n",
      "  libedit            conda-forge/linux-64::libedit-3.1.20191231-he28a2e2_2 \n",
      "  libglib            conda-forge/linux-64::libglib-2.78.0-hebfc3b9_0 \n",
      "  libiconv           conda-forge/linux-64::libiconv-1.17-h166bdaf_0 \n",
      "  libjpeg-turbo      conda-forge/linux-64::libjpeg-turbo-3.0.0-hd590300_1 \n",
      "  libpng             conda-forge/linux-64::libpng-1.6.39-h753d276_0 \n",
      "  libtiff            conda-forge/linux-64::libtiff-4.6.0-ha9c0a0a_2 \n",
      "  libwebp-base       conda-forge/linux-64::libwebp-base-1.3.2-hd590300_0 \n",
      "  libxcb             conda-forge/linux-64::libxcb-1.15-h0b41bf4_0 \n",
      "  openjdk            conda-forge/linux-64::openjdk-21.0.0-haa376d0_0 \n",
      "  pcre2              conda-forge/linux-64::pcre2-10.40-hc3806b6_0 \n",
      "  perl               conda-forge/linux-64::perl-5.32.1-4_hd590300_perl5 \n",
      "  pixman             conda-forge/linux-64::pixman-0.42.2-h59595ed_0 \n",
      "  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-h36c2ea0_1001 \n",
      "  xorg-fixesproto    conda-forge/linux-64::xorg-fixesproto-5.0-h7f98852_1002 \n",
      "  xorg-inputproto    conda-forge/linux-64::xorg-inputproto-2.3.2-h7f98852_1002 \n",
      "  xorg-kbproto       conda-forge/linux-64::xorg-kbproto-1.0.7-h7f98852_1002 \n",
      "  xorg-libice        conda-forge/linux-64::xorg-libice-1.1.1-hd590300_0 \n",
      "  xorg-libsm         conda-forge/linux-64::xorg-libsm-1.2.4-h7391055_0 \n",
      "  xorg-libx11        conda-forge/linux-64::xorg-libx11-1.8.7-h8ee46fc_0 \n",
      "  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.11-hd590300_0 \n",
      "  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.3-h7f98852_0 \n",
      "  xorg-libxext       conda-forge/linux-64::xorg-libxext-1.3.4-h0b41bf4_2 \n",
      "  xorg-libxfixes     conda-forge/linux-64::xorg-libxfixes-5.0.3-h7f98852_1004 \n",
      "  xorg-libxi         conda-forge/linux-64::xorg-libxi-1.7.10-h7f98852_0 \n",
      "  xorg-libxrender    conda-forge/linux-64::xorg-libxrender-0.9.11-hd590300_0 \n",
      "  xorg-libxt         conda-forge/linux-64::xorg-libxt-1.3.0-hd590300_1 \n",
      "  xorg-libxtst       conda-forge/linux-64::xorg-libxtst-1.2.3-h7f98852_1002 \n",
      "  xorg-recordproto   conda-forge/linux-64::xorg-recordproto-1.14.2-h7f98852_1002 \n",
      "  xorg-renderproto   conda-forge/linux-64::xorg-renderproto-0.11.1-h7f98852_1002 \n",
      "  xorg-xextproto     conda-forge/linux-64::xorg-xextproto-7.3.0-h0b41bf4_1003 \n",
      "  xorg-xproto        conda-forge/linux-64::xorg-xproto-7.0.31-h7f98852_1007 \n",
      "  zlib               conda-forge/linux-64::zlib-1.2.13-hd590300_5 \n",
      "  zstd               conda-forge/linux-64::zstd-1.5.5-hfc55251_0 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "libxcb-1.15          | 375 KB    |                                       |   0% \n",
      "giflib-5.2.1         | 76 KB     |                                       |   0% \u001b[A\n",
      "\n",
      "keyutils-1.6.1       | 115 KB    |                                       |   0% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "alsa-lib-1.2.10      | 542 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "pixman-0.42.2        | 376 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxext-1.3.4   | 49 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libpng-1.6.39        | 276 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxt-1.3.0     | 370 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-inputproto-2.3. | 19 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libdeflate-1.19      | 66 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxfixes-5.0.3 | 18 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-inconsolata | 94 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fonts-conda-forge-1  | 4 KB      |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxau-1.0.11   | 14 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxdmcp-1.1.3  | 19 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "zlib-1.2.13          | 91 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libx11-1.8.7    | 809 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libiconv-1.17        | 1.4 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libtiff-4.6.0        | 277 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxrender-0.9. | 37 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "libxcb-1.15          | 375 KB    | #5                                    |   4% \u001b[A\n",
      "\n",
      "keyutils-1.6.1       | 115 KB    | #####1                                |  14% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "alsa-lib-1.2.10      | 542 KB    | #                                     |   3% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "pixman-0.42.2        | 376 KB    | #5                                    |   4% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxext-1.3.4   | 49 KB     | ############                          |  33% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libpng-1.6.39        | 276 KB    | ##1                                   |   6% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "alsa-lib-1.2.10      | 542 KB    | ####################################  |  97% \u001b[A\u001b[A\u001b[A\n",
      "giflib-5.2.1         | 76 KB     | ##################################### | 100% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxt-1.3.0     | 370 KB    | #5                                    |   4% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "giflib-5.2.1         | 76 KB     | ##################################### | 100% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-inputproto-2.3. | 19 KB     | ##############################9       |  84% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxext-1.3.4   | 49 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libdeflate-1.19      | 66 KB     | #########                             |  24% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    | 3                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "pixman-0.42.2        | 376 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "pixman-0.42.2        | 376 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "keyutils-1.6.1       | 115 KB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "keyutils-1.6.1       | 115 KB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | #8                                    |   5% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | 1                                     |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    | #####5                                |  15% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libpng-1.6.39        | 276 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libpng-1.6.39        | 276 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-inconsolata | 94 KB     | ######2                               |  17% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxfixes-5.0.3 | 18 KB     | #################################4    |  90% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-inputproto-2.3. | 19 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libxcb-1.15          | 375 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fonts-conda-forge-1  | 4 KB      | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-inconsolata | 94 KB     | #########################1            |  68% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | 9                                     |   3% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libdeflate-1.19      | 66 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libdeflate-1.19      | 66 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    | ########6                             |  23% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxfixes-5.0.3 | 18 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ####                                  |  11% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "alsa-lib-1.2.10      | 542 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxt-1.3.0     | 370 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxt-1.3.0     | 370 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxau-1.0.11   | 14 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fonts-conda-forge-1  | 4 KB      | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxdmcp-1.1.3  | 19 KB     | ###############################6      |  86% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ####1                                 |  11% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-inconsolata | 94 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-inconsolata | 94 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxau-1.0.11   | 14 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxdmcp-1.1.3  | 19 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "zlib-1.2.13          | 91 KB     | ######5                               |  18% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ######8                               |  19% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #####7                                |  16% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libx11-1.8.7    | 809 KB    | 7                                     |   2% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libiconv-1.17        | 1.4 MB    | 4                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libx11-1.8.7    | 809 KB    | #######3                              |  20% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ########1                             |  22% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "zlib-1.2.13          | 91 KB     | ###################5                  |  53% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libiconv-1.17        | 1.4 MB    | ####5                                 |  12% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #######3                              |  20% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "zlib-1.2.13          | 91 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | #########3                            |  25% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libtiff-4.6.0        | 277 KB    | ##1                                   |   6% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #########7                            |  26% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libx11-1.8.7    | 809 KB    | ###########7                          |  32% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libiconv-1.17        | 1.4 MB    | #################5                    |  47% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | #############                         |  35% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ##############8                       |  40% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxrender-0.9. | 37 KB     | ################                      |  43% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libtiff-4.6.0        | 277 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libtiff-4.6.0        | 277 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ###############2                      |  41% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #################7                    |  48% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxrender-0.9. | 37 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libxrender-0.9. | 37 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | #################2                    |  47% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ####################3                 |  55% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libiconv-1.17        | 1.4 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libiconv-1.17        | 1.4 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ###################                   |  52% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libx11-1.8.7    | 809 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "xorg-libx11-1.8.7    | 809 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ######################8               |  62% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ######################7               |  61% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #########################             |  68% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ###########################2          |  74% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ########################8             |  67% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #############################5        |  80% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ###########################8          |  75% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ###############################6      |  86% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ##############################1       |  81% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ################################1     |  87% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | #################################5    |  91% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ##################################    |  92% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ####################################  |  98% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libcups-2.3.3        | 4.3 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "perl-5.32.1          | 12.7 MB   | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\u001b[A\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\u001b[A\n",
      "\n",
      "\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install fastqc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "571e0035-a7bd-402b-931e-fa488f9692fa",
   "metadata": {},
   "source": [
    "## to reach help file for fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5597271c-bc85-4c87-88cb-7d9b521f0a07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "            FastQC - A high throughput sequence QC analysis tool\n",
      "\n",
      "SYNOPSIS\n",
      "\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
      "\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
      "           [-c contaminant file] seqfile1 .. seqfileN\n",
      "\n",
      "DESCRIPTION\n",
      "\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\n",
      "    control report consisting of a number of different modules, each one of \n",
      "    which will help to identify a different potential type of problem in your\n",
      "    data.\n",
      "    \n",
      "    If no files to process are specified on the command line then the program\n",
      "    will start as an interactive graphical application.  If files are provided\n",
      "    on the command line then the program will run with no user interaction\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\n",
      "    analysis pipeline.\n",
      "    \n",
      "    The options for the program as as follows:\n",
      "    \n",
      "    -h --help       Print this help file and exit\n",
      "    \n",
      "    -v --version    Print the version of the program and exit\n",
      "    \n",
      "    -o --outdir     Create all output files in the specified output directory.\n",
      "                    Please note that this directory must exist as the program\n",
      "                    will not create it.  If this option is not set then the \n",
      "                    output file for each sequence file is created in the same\n",
      "                    directory as the sequence file which was processed.\n",
      "                    \n",
      "    --casava        Files come from raw casava output. Files in the same sample\n",
      "                    group (differing only by the group number) will be analysed\n",
      "                    as a set rather than individually. Sequences with the filter\n",
      "                    flag set in the header will be excluded from the analysis.\n",
      "                    Files must have the same names given to them by casava\n",
      "                    (including being gzipped and ending with .gz) otherwise they\n",
      "                    won't be grouped together correctly.\n",
      "                    \n",
      "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
      "                    this mode you can pass in directories to process and the program\n",
      "                    will take in all fast5 files within those directories and produce\n",
      "                    a single output file from the sequences found in all files.                    \n",
      "                    \n",
      "    --nofilter      If running with --casava then don't remove read flagged by\n",
      "                    casava as poor quality when performing the QC analysis.\n",
      "                   \n",
      "    --extract       If set then the zipped output file will be uncompressed in\n",
      "                    the same directory after it has been created. If --delete is \n",
      "                    also specified then the zip file will be removed after the \n",
      "                    contents are unzipped. \n",
      "                    \n",
      "    -j --java       Provides the full path to the java binary you want to use to\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\n",
      "                    your path.\n",
      "                   \n",
      "    --noextract     Do not uncompress the output file after creating it.  You\n",
      "                    should set this option if you do not wish to uncompress\n",
      "                    the output when running in non-interactive mode.\n",
      "                    \n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
      "                    show data for every base in the read.  WARNING: Using this\n",
      "                    option will cause fastqc to crash and burn if you use it on\n",
      "                    really long reads, and your plots may end up a ridiculous size.\n",
      "                    You have been warned!\n",
      "                    \n",
      "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
      "                    to be shown in the report.  As long as you set this to a value\n",
      "                    greater or equal to your longest read length then this will be\n",
      "                    the sequence length used to create your read groups.  This can\n",
      "                    be useful for making directly comaparable statistics from \n",
      "                    datasets with somewhat variable read lengths.\n",
      "\n",
      "    --dup_length    Sets a length to which the sequences will be truncated when \n",
      "                    defining them to be duplicates, affecting the duplication and\n",
      "                    overrepresented sequences plot.  This can be useful if you have\n",
      "                    long reads with higher levels of miscalls, or contamination with\n",
      "                    adapter dimers containing UMI sequences.\n",
      "\n",
      "                    \n",
      "    -f --format     Bypasses the normal sequence file format detection and\n",
      "                    forces the program to use the specified format.  Valid\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
      "                    \n",
      "\n",
      "    --memory        Sets the base amount of memory, in Megabytes, used to process \n",
      "                    each file.  Defaults to 512MB.  You may need to increase this if\n",
      "                    you have a file with very long sequences in it.\n",
      "                \n",
      "    --svg           Save the graphs in the report in SVG format.\n",
      "\n",
      "    -t --threads    Specifies the number of files which can be processed\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\n",
      "                    memory so you shouldn't run more threads than your\n",
      "                    available memory will cope with, and not more than\n",
      "                    6 threads on a 32 bit machine\n",
      "                  \n",
      "    -c              Specifies a non-default file which contains the list of\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\n",
      "                    The file must contain sets of named contaminants in the\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
      "                    be ignored.\n",
      "\n",
      "    -a              Specifies a non-default file which contains the list of\n",
      "    --adapters      adapter sequences which will be explicity searched against\n",
      "                    the library. The file must contain sets of named adapters\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
      "                    will be ignored.\n",
      "                    \n",
      "    -l              Specifies a non-default file which contains a set of criteria\n",
      "    --limits        which will be used to determine the warn/error limits for the\n",
      "                    various modules.  This file can also be used to selectively \n",
      "                    remove some modules from the output all together.  The format\n",
      "                    needs to mirror the default limits.txt file found in the\n",
      "                    Configuration folder.\n",
      "                    \n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\n",
      "                    length is 7 if not specified.\n",
      "                    \n",
      "   -q --quiet       Suppress all progress messages on stdout and only report errors.\n",
      "   \n",
      "   -d --dir         Selects a directory to be used for temporary files written when\n",
      "                    generating report images. Defaults to system temp directory if\n",
      "                    not specified.\n",
      "                    \n",
      "BUGS\n",
      "\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
      "                   \n",
      "    \n"
     ]
    }
   ],
   "source": [
    "!fastqc --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "28a33000-c043-48d2-b63e-df922ab856b4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/sequences'"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f2ae86b-3911-49e0-9ccc-2137e0276b9d",
   "metadata": {},
   "source": [
    "## run fastqc for all fastq files in the current directory and save into other directory (results folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "974b3dc0-4439-40e2-891c-52c5cb71f955",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null\n",
      "null\n",
      "Started analysis of SRR2589044_1.fastq\n",
      "Approx 5% complete for SRR2589044_1.fastq\n",
      "Approx 10% complete for SRR2589044_1.fastq\n",
      "Approx 15% complete for SRR2589044_1.fastq\n",
      "Approx 20% complete for SRR2589044_1.fastq\n",
      "Approx 25% complete for SRR2589044_1.fastq\n",
      "Approx 30% complete for SRR2589044_1.fastq\n",
      "Approx 35% complete for SRR2589044_1.fastq\n",
      "Approx 40% complete for SRR2589044_1.fastq\n",
      "Approx 45% complete for SRR2589044_1.fastq\n",
      "Approx 50% complete for SRR2589044_1.fastq\n",
      "Approx 55% complete for SRR2589044_1.fastq\n",
      "Approx 60% complete for SRR2589044_1.fastq\n",
      "Approx 65% complete for SRR2589044_1.fastq\n",
      "Approx 70% complete for SRR2589044_1.fastq\n",
      "Approx 75% complete for SRR2589044_1.fastq\n",
      "Approx 80% complete for SRR2589044_1.fastq\n",
      "Approx 85% complete for SRR2589044_1.fastq\n",
      "Approx 90% complete for SRR2589044_1.fastq\n",
      "Approx 95% complete for SRR2589044_1.fastq\n",
      "Analysis complete for SRR2589044_1.fastq\n",
      "Started analysis of SRR2589044_2.fastq\n",
      "Approx 5% complete for SRR2589044_2.fastq\n",
      "Approx 10% complete for SRR2589044_2.fastq\n",
      "Approx 15% complete for SRR2589044_2.fastq\n",
      "Approx 20% complete for SRR2589044_2.fastq\n",
      "Approx 25% complete for SRR2589044_2.fastq\n",
      "Approx 30% complete for SRR2589044_2.fastq\n",
      "Approx 35% complete for SRR2589044_2.fastq\n",
      "Approx 40% complete for SRR2589044_2.fastq\n",
      "Approx 45% complete for SRR2589044_2.fastq\n",
      "Approx 50% complete for SRR2589044_2.fastq\n",
      "Approx 55% complete for SRR2589044_2.fastq\n",
      "Approx 60% complete for SRR2589044_2.fastq\n",
      "Approx 65% complete for SRR2589044_2.fastq\n",
      "Approx 70% complete for SRR2589044_2.fastq\n",
      "Approx 75% complete for SRR2589044_2.fastq\n",
      "Approx 80% complete for SRR2589044_2.fastq\n",
      "Approx 85% complete for SRR2589044_2.fastq\n",
      "Approx 90% complete for SRR2589044_2.fastq\n",
      "Approx 95% complete for SRR2589044_2.fastq\n",
      "Analysis complete for SRR2589044_2.fastq\n"
     ]
    }
   ],
   "source": [
    "!fastqc *.fastq -O ../results/fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "62560b38-ede0-40bf-be4d-a263b2e3e5b2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/sequences'"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "8abd36f9-864e-4355-8e40-66226ca89c3a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results/fastqc\n"
     ]
    }
   ],
   "source": [
    "cd ../results/fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "debbabf0-99e0-4fb9-aa30-4f095d9c170b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044_1_fastqc.html\u001b[0m*  \u001b[01;32mSRR2589044_2_fastqc.html\u001b[0m*\n",
      "\u001b[01;32mSRR2589044_1_fastqc.zip\u001b[0m*   \u001b[01;32mSRR2589044_2_fastqc.zip\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3112b51b-2d3c-430c-8c73-35b9339bf04b",
   "metadata": {},
   "source": [
    "## Copying Illumina ADAPTOR sequences to squences folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "51e7f5ea-f27d-4f5f-9cb5-d9ba0197c42e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis\n"
     ]
    }
   ],
   "source": [
    "cd ../.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "15552cf0-0ce9-4b04-967d-cab5088579b6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mNGS_DNAseq_analysis.ipynb\u001b[0m*  \u001b[01;32mNexteraPE-PE.fa\u001b[0m*  \u001b[34;42mreference\u001b[0m/  \u001b[34;42mresults\u001b[0m/  \u001b[34;42msequences\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "38e25dbd-8998-4d55-8866-95f7dd07ea70",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "cp NexteraPE-PE.fa sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e5d79783-7db5-4610-9a12-2ce1c3a352b9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc0e19a3-b668-4cc1-a8d9-54f32ceb59fb",
   "metadata": {},
   "source": [
    "## Trimming reads with trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "88a6225b-7e1f-4b73-a905-2c9c8d2a1773",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/sequences\n"
     ]
    }
   ],
   "source": [
    "cd sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8a7c4b47-fac8-46ff-932d-2033ea9cb880",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mNexteraPE-PE.fa\u001b[0m*  \u001b[01;32mSRR2589044_1.fastq\u001b[0m*  \u001b[01;32mSRR2589044_2.fastq\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3566a01f-cc83-4e32-a864-eb9a62cc6f6b",
   "metadata": {},
   "source": [
    "## Install trimmomatic tool for trimming fasta sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "31ed9627-f040-4711-bde6-1cc363fd4821",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.7.4\n",
      "  latest version: 23.9.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.9.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/sdogan/anaconda3/envs/NGS_DNAseq_analysis\n",
      "\n",
      "  added / updated specs:\n",
      "    - trimmomatic\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         144 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "                                                                                \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ba29f436-d61c-495e-9733-199a98a0ff2a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mNexteraPE-PE.fa\u001b[0m*  \u001b[01;32mSRR2589044_1.fastq\u001b[0m*  \u001b[01;32mSRR2589044_2.fastq\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "29438e90-c070-4f7e-934a-1a73e46c507f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticPE: Started with arguments:\n",
      " SRR2589044_1.fastq SRR2589044_2.fastq ../results/trimmomatic/SRR2589044_1_paired.fastq.gz ../results/trimmomatic/SRR2589044_1_unpaired.fastq.gz ../results/trimmomatic/SRR2589044_2_paired.fastq.gz ../results/trimmomatic/SRR2589044_2_unpaired.fastq.gz SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15\n",
      "Multiple cores found: Using 4 threads\n",
      "Using PrefixPair: 'AGATGTGTATAAGAGACAG' and 'AGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTCCGAGCCCACGAGAC'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTGACGCTGCCGACGA'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 1107090 Both Surviving: 885220 (79.96%) Forward Only Surviving: 216472 (19.55%) Reverse Only Surviving: 2850 (0.26%) Dropped: 2548 (0.23%)\n",
      "TrimmomaticPE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "!trimmomatic PE SRR2589044_1.fastq SRR2589044_2.fastq\\\n",
    "../results/trimmomatic/SRR2589044_1_paired.fastq.gz ../results/trimmomatic/SRR2589044_1_unpaired.fastq.gz\\\n",
    "../results/trimmomatic/SRR2589044_2_paired.fastq.gz ../results/trimmomatic/SRR2589044_2_unpaired.fastq.gz\\\n",
    "SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "25877436-f55e-4fd7-aa65-a169db90a357",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results/trimmomatic\n"
     ]
    }
   ],
   "source": [
    "cd ../results/trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "02def3ee-73b8-43f3-b2db-50b792ef000d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044_1_paired.fastq.gz\u001b[0m*    \u001b[01;32mSRR2589044_2_paired.fastq.gz\u001b[0m*\n",
      "\u001b[01;32mSRR2589044_1_unpaired.fastq.gz\u001b[0m*  \u001b[01;32mSRR2589044_2_unpaired.fastq.gz\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9594ad2c-beb0-4f3e-b914-e82e43fad005",
   "metadata": {},
   "source": [
    "## Unzipping files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "4e84cdb4-c943-49b9-911a-3ceb559d5c72",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!gunzip *.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "bc69dbd8-574a-4020-8a8f-4139b472131a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044_1_paired.fastq\u001b[0m*    \u001b[01;32mSRR2589044_2_paired.fastq\u001b[0m*\n",
      "\u001b[01;32mSRR2589044_1_unpaired.fastq\u001b[0m*  \u001b[01;32mSRR2589044_2_unpaired.fastq\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4085a2cf-4ae0-484d-93b9-49f88079e8d6",
   "metadata": {},
   "source": [
    "## Download reference genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "c4dd3c1b-10a4-4d91-bcd3-f2478d1b020f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis\n"
     ]
    }
   ],
   "source": [
    "cd ../.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "b688dd74-e96c-44a0-85b7-70f38b4a708d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/reference\n"
     ]
    }
   ],
   "source": [
    "cd reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "ec9f7179-86f2-426e-8551-2b743d0ec88f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 1343k  100 1343k    0     0   357k      0  0:00:03  0:00:03 --:--:--  357k\n"
     ]
    }
   ],
   "source": [
    "!curl -L ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/017/985/GCA_000017985.1_ASM1798v1/GCA_000017985.1_ASM1798v1_genomic.fna.gz -o ecoli_rel606.fasta.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7498c0b8-0fbb-4171-9f12-d0802144242a",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Indexing reference genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "1fe37202-3249-4ef3-8987-35f5c5ba0778",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/reference'"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "17f8bd98-dc43-4e93-8157-281aae6ad935",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mecoli_rel606.fasta.gz\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "f66a595b-0596-498b-9c87-6ff6edadca48",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!gunzip *.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "bf02d49e-a9c1-4d55-90a9-159c0f8cf1db",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mecoli_rel606.fasta\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cdee032d-c1cc-4a9e-bed0-b4fddc968ab4",
   "metadata": {},
   "source": [
    "## Install bwa indexing tool for fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "06ad8910-691d-48fa-87d9-1cc4cf849f40",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.7.4\n",
      "  latest version: 23.9.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.9.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/sdogan/anaconda3/envs/NGS_DNAseq_analysis\n",
      "\n",
      "  added / updated specs:\n",
      "    - bwa\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    bwa-0.7.17                 |      he4a0461_11         192 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         192 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  bwa                bioconda/linux-64::bwa-0.7.17-he4a0461_11 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "                                                                                \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install bwa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "c1977eff-f9f3-4802-8104-69d44f0131f8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 0.05 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[bwa_index] 0.88 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.03 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.02 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 0.28 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index ecoli_rel606.fasta\n",
      "[main] Real time: 1.283 sec; CPU: 1.281 sec\n"
     ]
    }
   ],
   "source": [
    "!bwa index ecoli_rel606.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "247b06ae-48c8-47cd-8781-ea73fbbe503e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mecoli_rel606.fasta\u001b[0m*      \u001b[01;32mecoli_rel606.fasta.ann\u001b[0m*  \u001b[01;32mecoli_rel606.fasta.pac\u001b[0m*\n",
      "\u001b[01;32mecoli_rel606.fasta.amb\u001b[0m*  \u001b[01;32mecoli_rel606.fasta.bwt\u001b[0m*  \u001b[01;32mecoli_rel606.fasta.sa\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "e92c11c8-d90f-4bad-a8e9-eaea71f7fdb6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd18e10a-b085-45ae-8879-acc04c975d92",
   "metadata": {},
   "source": [
    "## Aligning reads cleared by trimmomatic to reference genome - generates sam files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "2471b0fe-68af-42bb-8e82-9293a8531077",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 77804 sequences (10000111 bp)...\n",
      "[M::process] read 74194 sequences (10000037 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (376, 35818, 10, 389)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 731, 1775)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4611)\n",
      "[M::mem_pestat] mean and std.dev: (1015.53, 988.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6029)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (284, 501, 709)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1559)\n",
      "[M::mem_pestat] mean and std.dev: (512.26, 257.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1984)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (249, 897, 2391)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6675)\n",
      "[M::mem_pestat] mean and std.dev: (1304.20, 1285.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8817)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (411, 821, 1941)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5001)\n",
      "[M::mem_pestat] mean and std.dev: (1106.63, 1049.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6531)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 77804 reads in 2.609 CPU sec, 2.530 real sec\n",
      "[M::process] read 75440 sequences (10000149 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (360, 34097, 18, 365)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (389, 734, 1641)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4145)\n",
      "[M::mem_pestat] mean and std.dev: (935.84, 869.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5397)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (284, 502, 711)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1565)\n",
      "[M::mem_pestat] mean and std.dev: (513.04, 256.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1992)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (614, 1246, 2207)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5393)\n",
      "[M::mem_pestat] mean and std.dev: (1502.33, 1223.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6986)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (389, 752, 1879)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4859)\n",
      "[M::mem_pestat] mean and std.dev: (1084.10, 1063.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6349)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74194 reads in 4.656 CPU sec, 4.551 real sec\n",
      "[M::process] read 76110 sequences (10000047 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (348, 34932, 8, 340)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (401, 804, 2038)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5312)\n",
      "[M::mem_pestat] mean and std.dev: (1222.19, 1213.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6949)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (280, 500, 711)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1573)\n",
      "[M::mem_pestat] mean and std.dev: (512.91, 258.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2004)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (341, 698, 2223)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5987)\n",
      "[M::mem_pestat] mean and std.dev: (1241.59, 1369.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7869)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 75440 reads in 4.641 CPU sec, 4.357 real sec\n",
      "[M::process] read 74180 sequences (10000292 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (351, 35158, 7, 329)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (365, 782, 1903)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4979)\n",
      "[M::mem_pestat] mean and std.dev: (1143.92, 1164.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6517)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (288, 520, 730)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1614)\n",
      "[M::mem_pestat] mean and std.dev: (527.08, 265.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2056)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (384, 745, 1997)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5223)\n",
      "[M::mem_pestat] mean and std.dev: (1165.53, 1256.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6836)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 76110 reads in 4.688 CPU sec, 4.497 real sec\n",
      "[M::process] read 74330 sequences (10000129 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (355, 34256, 11, 301)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (383, 729, 1832)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4730)\n",
      "[M::mem_pestat] mean and std.dev: (1092.53, 1072.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6179)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (295, 525, 737)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1621)\n",
      "[M::mem_pestat] mean and std.dev: (533.79, 267.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2063)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (450, 651, 2256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5868)\n",
      "[M::mem_pestat] mean and std.dev: (1029.80, 988.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7674)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (401, 773, 1928)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4982)\n",
      "[M::mem_pestat] mean and std.dev: (1128.53, 1096.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6509)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74180 reads in 4.672 CPU sec, 4.518 real sec\n",
      "[M::process] read 73158 sequences (10000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (351, 34303, 6, 362)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (405, 722, 1659)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4167)\n",
      "[M::mem_pestat] mean and std.dev: (979.02, 917.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5421)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (304, 542, 749)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1639)\n",
      "[M::mem_pestat] mean and std.dev: (544.99, 268.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2084)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (380, 704, 1840)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4760)\n",
      "[M::mem_pestat] mean and std.dev: (1055.35, 1076.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6220)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74330 reads in 4.547 CPU sec, 4.376 real sec\n",
      "[M::process] read 73808 sequences (10000120 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (384, 33674, 10, 420)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (460, 894, 2200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5680)\n",
      "[M::mem_pestat] mean and std.dev: (1247.40, 1220.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7420)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (375, 631, 817)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1701)\n",
      "[M::mem_pestat] mean and std.dev: (611.34, 279.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2143)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (402, 1308, 2182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5742)\n",
      "[M::mem_pestat] mean and std.dev: (1532.80, 1363.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7522)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (407, 769, 2189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5753)\n",
      "[M::mem_pestat] mean and std.dev: (1271.84, 1286.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7535)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 73158 reads in 4.859 CPU sec, 4.693 real sec\n",
      "[M::process] read 74618 sequences (10000130 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (456, 33751, 13, 428)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (494, 827, 2056)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5180)\n",
      "[M::mem_pestat] mean and std.dev: (1245.38, 1172.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6742)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (386, 640, 828)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1712)\n",
      "[M::mem_pestat] mean and std.dev: (619.26, 279.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2154)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (256, 809, 3150)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8938)\n",
      "[M::mem_pestat] mean and std.dev: (1488.00, 1402.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (468, 817, 2158)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5538)\n",
      "[M::mem_pestat] mean and std.dev: (1352.86, 1300.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7228)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 73808 reads in 4.922 CPU sec, 4.721 real sec\n",
      "[M::process] read 77738 sequences (10000133 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (404, 34291, 13, 416)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (452, 892, 2264)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5888)\n",
      "[M::mem_pestat] mean and std.dev: (1369.36, 1347.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7700)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 636, 822)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1712)\n",
      "[M::mem_pestat] mean and std.dev: (613.71, 279.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2157)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 624, 1266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3380)\n",
      "[M::mem_pestat] mean and std.dev: (756.42, 826.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4437)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (431, 847, 2234)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5840)\n",
      "[M::mem_pestat] mean and std.dev: (1235.20, 1242.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7643)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74618 reads in 4.750 CPU sec, 4.578 real sec\n",
      "[M::process] read 73398 sequences (10000206 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (401, 35725, 8, 433)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (461, 858, 2295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5963)\n",
      "[M::mem_pestat] mean and std.dev: (1378.41, 1418.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7797)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (360, 622, 813)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1719)\n",
      "[M::mem_pestat] mean and std.dev: (601.21, 278.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2172)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (461, 898, 2294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5960)\n",
      "[M::mem_pestat] mean and std.dev: (1349.32, 1292.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7793)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 77738 reads in 4.922 CPU sec, 4.734 real sec\n",
      "[M::process] read 73948 sequences (10000131 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (402, 33728, 7, 384)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (404, 803, 1884)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4844)\n",
      "[M::mem_pestat] mean and std.dev: (1113.73, 1055.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6324)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (372, 623, 810)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1686)\n",
      "[M::mem_pestat] mean and std.dev: (604.41, 275.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2124)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (420, 850, 2325)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6135)\n",
      "[M::mem_pestat] mean and std.dev: (1360.93, 1402.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8040)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 73398 reads in 4.938 CPU sec, 4.763 real sec\n",
      "[M::process] read 74446 sequences (10000038 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (400, 34005, 11, 386)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (510, 860, 2147)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5421)\n",
      "[M::mem_pestat] mean and std.dev: (1220.78, 1136.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7058)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (392, 647, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1709)\n",
      "[M::mem_pestat] mean and std.dev: (624.02, 280.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2148)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (800, 1467, 2635)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6305)\n",
      "[M::mem_pestat] mean and std.dev: (1544.73, 1293.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8140)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (491, 883, 2116)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5366)\n",
      "[M::mem_pestat] mean and std.dev: (1243.74, 1192.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6991)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 73948 reads in 4.734 CPU sec, 4.464 real sec\n",
      "[M::process] read 75630 sequences (10000036 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (444, 34103, 12, 431)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (455, 813, 1936)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4898)\n",
      "[M::mem_pestat] mean and std.dev: (1116.26, 1038.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6379)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (383, 640, 824)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1706)\n",
      "[M::mem_pestat] mean and std.dev: (617.67, 280.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2147)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (323, 1188, 1907)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5075)\n",
      "[M::mem_pestat] mean and std.dev: (1217.17, 1123.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6659)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (435, 799, 1770)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4440)\n",
      "[M::mem_pestat] mean and std.dev: (1087.36, 1024.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5775)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74446 reads in 5.000 CPU sec, 4.697 real sec\n",
      "[M::process] read 75956 sequences (10000054 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (406, 34705, 13, 443)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (479, 845, 1920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4802)\n",
      "[M::mem_pestat] mean and std.dev: (1160.75, 1041.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6243)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (374, 629, 817)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1703)\n",
      "[M::mem_pestat] mean and std.dev: (609.03, 277.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2146)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (99, 526, 5551)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16455)\n",
      "[M::mem_pestat] mean and std.dev: (2106.31, 2696.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21907)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 875, 2202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5710)\n",
      "[M::mem_pestat] mean and std.dev: (1306.64, 1291.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7464)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 75630 reads in 4.359 CPU sec, 4.148 real sec\n",
      "[M::process] read 73006 sequences (10000214 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (444, 34815, 13, 457)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (457, 820, 2124)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5458)\n",
      "[M::mem_pestat] mean and std.dev: (1278.34, 1227.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7125)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (384, 643, 832)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1728)\n",
      "[M::mem_pestat] mean and std.dev: (620.52, 282.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2176)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (371, 1263, 4297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12149)\n",
      "[M::mem_pestat] mean and std.dev: (2190.23, 2095.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16075)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (431, 731, 1852)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4694)\n",
      "[M::mem_pestat] mean and std.dev: (1056.30, 993.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6115)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 75956 reads in 4.609 CPU sec, 4.407 real sec\n",
      "[M::process] read 73504 sequences (10000024 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (409, 33522, 10, 419)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (443, 813, 2154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5576)\n",
      "[M::mem_pestat] mean and std.dev: (1280.39, 1273.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7287)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (409, 668, 854)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1744)\n",
      "[M::mem_pestat] mean and std.dev: (643.77, 288.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2189)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (573, 1444, 2546)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6492)\n",
      "[M::mem_pestat] mean and std.dev: (1242.67, 1015.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8465)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (459, 829, 1970)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4992)\n",
      "[M::mem_pestat] mean and std.dev: (1191.76, 1147.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6503)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 73006 reads in 4.609 CPU sec, 4.485 real sec\n",
      "[M::process] read 74392 sequences (10000075 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (399, 33639, 12, 435)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (512, 864, 2092)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5252)\n",
      "[M::mem_pestat] mean and std.dev: (1325.04, 1252.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (410, 666, 854)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1742)\n",
      "[M::mem_pestat] mean and std.dev: (644.01, 289.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2186)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1048, 2267, 4259)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10681)\n",
      "[M::mem_pestat] mean and std.dev: (2759.83, 2500.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13892)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (507, 901, 2466)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6384)\n",
      "[M::mem_pestat] mean and std.dev: (1400.55, 1343.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8343)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 73504 reads in 4.734 CPU sec, 4.521 real sec\n",
      "[M::process] read 76402 sequences (10000146 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (414, 34186, 11, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (467, 875, 2020)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5126)\n",
      "[M::mem_pestat] mean and std.dev: (1232.11, 1108.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6679)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (385, 650, 840)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1750)\n",
      "[M::mem_pestat] mean and std.dev: (627.20, 287.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2205)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (450, 498, 3164)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8592)\n",
      "[M::mem_pestat] mean and std.dev: (1769.36, 2178.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11306)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (513, 994, 2195)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5559)\n",
      "[M::mem_pestat] mean and std.dev: (1305.49, 1169.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7241)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74392 reads in 4.516 CPU sec, 4.249 real sec\n",
      "[M::process] read 75066 sequences (10000065 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (448, 34932, 18, 464)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (456, 849, 2127)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5469)\n",
      "[M::mem_pestat] mean and std.dev: (1213.86, 1169.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7140)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (376, 641, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1735)\n",
      "[M::mem_pestat] mean and std.dev: (617.47, 283.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2188)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (178, 1165, 1742)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4870)\n",
      "[M::mem_pestat] mean and std.dev: (1154.71, 1052.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6434)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (419, 827, 1828)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4646)\n",
      "[M::mem_pestat] mean and std.dev: (1132.32, 1085.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6055)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 76402 reads in 4.797 CPU sec, 4.610 real sec\n",
      "[M::process] read 74224 sequences (10000019 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (433, 34367, 9, 451)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (432, 833, 2032)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5232)\n",
      "[M::mem_pestat] mean and std.dev: (1177.13, 1117.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (393, 649, 835)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1719)\n",
      "[M::mem_pestat] mean and std.dev: (626.38, 282.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2161)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (488, 968, 2628)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6908)\n",
      "[M::mem_pestat] mean and std.dev: (1532.16, 1551.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9048)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 75066 reads in 4.656 CPU sec, 4.463 real sec\n",
      "[M::process] read 74342 sequences (10000086 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (432, 34056, 10, 414)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (448, 810, 1827)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4585)\n",
      "[M::mem_pestat] mean and std.dev: (1071.18, 1000.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5964)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (399, 658, 850)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1752)\n",
      "[M::mem_pestat] mean and std.dev: (637.51, 290.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2203)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (700, 3445, 4157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11071)\n",
      "[M::mem_pestat] mean and std.dev: (2647.70, 1664.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14528)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (480, 897, 2306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5958)\n",
      "[M::mem_pestat] mean and std.dev: (1295.66, 1219.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7784)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74224 reads in 5.031 CPU sec, 4.827 real sec\n",
      "[M::process] read 75672 sequences (10000011 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (400, 34142, 8, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (473, 897, 2036)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5162)\n",
      "[M::mem_pestat] mean and std.dev: (1221.37, 1130.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6725)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (407, 660, 845)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1721)\n",
      "[M::mem_pestat] mean and std.dev: (637.06, 286.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2159)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (418, 785, 2154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5626)\n",
      "[M::mem_pestat] mean and std.dev: (1251.32, 1265.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7362)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 74342 reads in 4.891 CPU sec, 4.810 real sec\n",
      "[M::process] read 49074 sequences (6169704 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (446, 34606, 14, 428)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (436, 891, 2284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5980)\n",
      "[M::mem_pestat] mean and std.dev: (1364.12, 1378.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7828)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 639, 828)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1730)\n",
      "[M::mem_pestat] mean and std.dev: (616.76, 284.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2181)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (391, 1349, 2039)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5335)\n",
      "[M::mem_pestat] mean and std.dev: (1428.14, 1206.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6983)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (413, 786, 1991)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5147)\n",
      "[M::mem_pestat] mean and std.dev: (1202.19, 1189.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6725)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 75672 reads in 4.922 CPU sec, 4.736 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (272, 22437, 7, 278)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (428, 728, 1787)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4505)\n",
      "[M::mem_pestat] mean and std.dev: (1015.14, 924.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5864)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (357, 617, 812)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1722)\n",
      "[M::mem_pestat] mean and std.dev: (598.92, 279.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2177)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (542, 998, 3001)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7919)\n",
      "[M::mem_pestat] mean and std.dev: (1880.74, 1899.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10378)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 49074 reads in 3.062 CPU sec, 2.903 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem reference/ecoli_rel606.fasta results/trimmomatic/SRR2589044_1_paired.fastq results/trimmomatic/SRR2589044_2_paired.fastq\n",
      "[main] Real time: 105.955 sec; CPU: 110.344 sec\n"
     ]
    }
   ],
   "source": [
    "!bwa mem reference/ecoli_rel606.fasta \\\n",
    "results/trimmomatic/SRR2589044_1_paired.fastq \\\n",
    "results/trimmomatic/SRR2589044_2_paired.fastq \\\n",
    "> results/sam/SRR2589044.aligned.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c527b1e1-ebda-4652-b03c-61a7791a88d4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results/sam\n"
     ]
    }
   ],
   "source": [
    "cd results/sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "a9ba0cac-e4e1-48ca-ad5a-b0480df5a435",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044.aligned.sam\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a1fe93d-be5c-415b-b333-f8ceb111e392",
   "metadata": {},
   "source": [
    "## Checking SAM files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c568964b-8b88-4218-8aaa-bf10d983bebd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SQ\tSN:CP000819.1\tLN:4629812\n",
      "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa mem reference/ecoli_rel606.fasta results/trimmomatic/SRR2589044_1_paired.fastq results/trimmomatic/SRR2589044_2_paired.fastq\n",
      "SRR2589044.5\t99\tCP000819.1\t296669\t60\t87M\t=\t297499\t872\tTTAGCGGTGACACGAGGCTGGAGGAATTGGGGATTAGCGAACGTTTGCAGCAGGAATGCAATGCCGTTGTTTGGGATCCGGCGAAAG\t@C@FFFFDFDHHHG1CFHHIHDGEHIEHIGE06FHE@BFAGAEHFEBDED@>ACDBAC>@5;5?:=B=@<A9?@02><?AB359><@\tNM:i:0\tMD:Z:87\tMC:Z:42M\tAS:i:87\tXS:i:0\n",
      "SRR2589044.5\t147\tCP000819.1\t297499\t60\t42M\t=\t296669\t-872\tTCACCGTTCGGATACAATGACCAGTTCCTGGTCAGGTATATC\t600EF?EFE@?AC?FEC<333BDFF@=GABDB<FDDDDD?@?\tNM:i:0\tMD:Z:42\tMC:Z:87M\tAS:i:42\tXS:i:0\n",
      "SRR2589044.6\t83\tCP000819.1\t4048710\t60\t150M\t=\t4048675\t-185\tGCCGGGCGCTAACGGGATACGCGGGATCACCAGGCAATGCAGCAGGTGATAAATATCCTGTTTGCCGTCCCAGACTTTGGTGGTGACCTTATTGTCCGCATCCAGCTCCTGCCACCAGGAACCATTTTCATAGTCCATCAGGTACTTAAT\t@>@5@DBC?BBBDCDA@>9>BBCDDDDCCDDDDDCEDCDDDBDDCCCCCCDDCDDDDDDDBDB<6DDDDDDEDCDCDDBFEFFCEEGHEFGHBEJIEHGHEGGHIGFBIEIFIHIJJJJIJIJJIJJIJJJIIIJJJFHHGHFDFFFCCC\tNM:i:0\tMD:Z:150\tMC:Z:150M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.6\t163\tCP000819.1\t4048675\t60\t150M\t=\t4048710\t185\tATATCCAGCAGACCCGCTGCAACCGCTGGAGCCATGCCGGGCGCTAACGGGATACGCGGGATCACCAGGCAATGCAGCAGGTGATAAATATCCTGTTTGCCGTCCCAGACTTTGGTGGTGACCTTATTGTCCGCATCCAGCTCCTGCCAC\tCCCFFFFFHHHHHJJJJJ>HIIIJIIIIIIJJIJJJGIJJIIGBDDDEDDDDDDDBBDDB;<BBDCDDDDDDDDDCCCBDDACDDEDDEDDCDECDDD?A>>9@DDDD<?CDDDDDD@BDDDCDCCC@CCDDD@>BDDDDDCDCCCCDDD\tNM:i:0\tMD:Z:150\tMC:Z:150M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.8\t83\tCP000819.1\t3197147\t60\t150M\t=\t3196712\t-585\tGCAAAGTGGTTAGCCAGATAAGAAAGCACTGGAATATTCTGCGCTTTGGCTTCCGCCATGTTGGCCGGAGACAGAGTAAACAGGCAGCTAAAGGCAAAGAACATCACCACTGCAACCATCAGCATGCTGGCACGAGAAATGATTTGGGAA\tCCDDDDDDDDDDDDEDDDDDDDCDEDDCDEEEEEDDDDDDDDDDDDDBABDDDDCDDDDDDDDDDFFFFHHHHHHJJJJIJJJJJJIHJIIIIIJJJJIJJJJIHDIHFIJJJGIJJIGJJIJJJJJIJIIG?JJJJHHHHHFFFFFCCC\tNM:i:0\tMD:Z:150\tMC:Z:150M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.8\t163\tCP000819.1\t3196712\t60\t150M\t=\t3197147\t585\tGTTACGGATTAAAACAGTTTGTATACGATGTTCAGGATGGTCAGCAGACCAATCACGGTAACAAACACGTTATCCAGACGACCACGGTATTTCGCCAGAGACGGCGCTTTACGGATGGCATACATCGGCAACAGGCACAGCAGGGATGCG\tBBCFFFFDHHHHHJJJJJJJIJIJJJJJJGHIJJJJIJHJHIJJJEECIIHIJGHIIIGIJGGGIEHHGFFFFEDDACCBDDBDDDDBBDEEDCBDDDBDD9@DDDDDDDDDD@BBB@CDDDDDDDDDDDBBDBDDBDDCDC@BBBCDBD\tNM:i:0\tMD:Z:150\tMC:Z:150M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.9\t83\tCP000819.1\t2687346\t60\t150M\t=\t2687343\t-153\tCGAAGTATGGCATCGAAGATTACTTAGAAATCAAATATATGTGCATCGGTCTTTAACTGGAGAACGCGAATGAGCAGCAATAAAGAGTTAATGCAGCGCCGCAGTCAGGCGATTCCCCGTGGTGTTGGGCAAATTCACCCGATTTTCGCT\tDDDDDCDDDDEEDDEDDEDCEDCEEECEDDEEDDDDEDDDDDBDDDDCCADCCDDDDDDDDDDDDDEDDDCDDDDDDDDDDDDDDDDDDEDDBBD@@DFFHHHHHIJIJIJIJJIIJIJJJJJJIIJJJJJIIJHGEHHHHGFFFFFCCC\tNM:i:0\tMD:Z:150\tMC:Z:150M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.9\t163\tCP000819.1\t2687343\t60\t150M\t=\t2687346\t153\tGTTCGAAGTATGGCATCGAAGATTACTTAGAAATCAAATATATGTGCATCGGTCTTTAACTGGAGAACGCGAATGAGCAGCAATAAAGAGTTAATGCAGCGCCGCAGTCAGGCGATTCCCCGTGGTGTTGGGCAAATTCACCCGATTTTC\t@CCFFFFFFFHHGIJJJIJJGIJJJJJJJIIJJIIJJIIJIJIEFIIJIJJIHHJGCHIIIIIJGIJJIHFFDDDDDDDDBDDDDDDDDDDEDEEDDCDDD>BDDDDDDDDDDDDDDDDDDDB@BDDDB@?BBDBBDDDA@CDDDB?BDE\tNM:i:0\tMD:Z:150\tMC:Z:150M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.10\t99\tCP000819.1\t1929127\t60\t150M\t=\t1930051\t973\tCGTCCGCACGATCCGGATCGAAAAATACCTGCACGATACCTTCGCGGTCGCGCATATCGATGAAGATCAGGCTACCAAGATCACGACGACGGTTGACCCAACCACACAGAGTCACCTGCTGCCCCACGTGGGACAAACGGAGCTGTCCAC\tB@<BDDDDHDFFFEEGIGGGEGIIIGIIIIIHIIGGCEGIFHHAHHFFCDDB?BBBCD@BDEDDDCDCCCA9<CC?CDDDDDDCB@@>@>BBBB>CCCDD@@B@BDBDBDDDDDDCCCDCDDBDDBB><AB<?BDDDBB@DB@BDDCDDD\tNM:i:0\tMD:Z:150\tMC:Z:49M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.10\t147\tCP000819.1\t1930051\t60\t49M\t=\t1929127\t-973\tGATTGCCGAAGACGCCTGTAGTGCCGCCAGCGCCGAGCAGCACAATAAC\t:(;;89.6=994>FCGGCGGGE>BA=:@HE?B@FE>C8DB?B8DDD@@@\tNM:i:0\tMD:Z:49\tMC:Z:150M\tAS:i:49\tXS:i:0\n",
      "SRR2589044.11\t83\tCP000819.1\t1336712\t60\t150M\t=\t1336067\t-795\tATCTGCAAAACCTACAACCCGGCGCGACGGTTCCGGTGACGCTTACGCGCGCGGATGGTAGCCAGGAAGTCGTACCCTGCCGTTGTCGTATCGACACCGCGACGGAGTTGACCTACTACCAGAACGACGGCATTTTGCATTATGTCATTC\t@444:2@C@:CC>9D?9BB>D@9<.98<?99>@8C?<?89DDDB@B@95.BBA?>@C<@??CC@DDDDDB@@>9BDDD@:@DDDBD@AADFFFFHHIHGFFAIJJIHFDB>GIGDDD?JIGHE?GHGGIJJJIIIGIHFDHHFFFFFCCC\tNM:i:0\tMD:Z:150\tMC:Z:145M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.11\t163\tCP000819.1\t1336067\t60\t145M\t=\t1336712\t795\tGAATTAACGTCACACGATCCGATACCTACGGTTGGCAGGAGGACTCAACCTATATTCGCTTATCGCCTTTCTTTGATGAAATGCAGGCAACACCAGCACCAGTGGAAGATATTCACGGTGCGCGGATCCTCGCAATGCTGGGGGA\t@@@FFFFFFFDFHGJIIHIGCGGGHIHJIIGIG:DHGHIB;<GEHJGH@E>CHHHFFDEBDDC@AD?DBCCDDDCDCC>:>ACDDD<0<BBBBBB?<CBB?8@AA>:?<>>>:::A?BD?@>9@B9>BD>CDDB&9CACCC?BBB\tNM:i:0\tMD:Z:145\tMC:Z:150M\tAS:i:145\tXS:i:0\n",
      "SRR2589044.12\t83\tCP000819.1\t2339660\t60\t81M\t=\t2338844\t-897\tGGCGCTTGTATTCCCCCATTTCGAAATCCGGGGTTAACTGAATCGCCGTGGTCGGACAGGCTTCTTCGCACAGACCACAGA\t<5(?:>CC>@BB@BBCAA?B?<>AB?;@?DFFDGHEEEHHEA@F@BFF<BGHGEGF>HDHCGBBDGHE>FAHAEDDBD@@@\tNM:i:0\tMD:Z:81\tMC:Z:66M\tAS:i:81\tXS:i:0\n"
     ]
    }
   ],
   "source": [
    "!head -n 15 SRR2589044.aligned.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "eb2ef684-f43f-4950-be1e-55fc45b2632a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR2589044.1107082\t163\tCP000819.1\t655255\t60\t145M\t=\t655388\t224\tCATCGCGACGCAGCGCTTTCTGATCTTCAGTCAGCGCATCAACGTTCAGTGCCGCAACATCACCTTTTGCTGTGTGCTCGTAAACCAGTTTCCAGACACGTTTCAGGAAGCGGTTAGCCCCTTCCACACCGGATTCCTGCCATTC\tCCCFDFFFHHHA?FEHGIIGHIIIIGIIIIIIIHHAHIIIHHHHFFFFFEDEEDDDDDBCDCDDDDDD>C>AC?CAC:@@9BDDCDDB@CEDDD@CDBC>A<CC@@CDD8A>BBB7(:@CD8ACCCD@@B5<9<<>C:3:ACCDB\tNM:i:0\tMD:Z:145\tMC:Z:91M\tAS:i:145\tXS:i:0\n",
      "SRR2589044.1107083\t99\tCP000819.1\t1906108\t60\t150M\t=\t1906756\t757\tATGGTAGACCAACGAAACCGGAACTTCGACGCCATCACGGGCGACTATCCACAGGTGTTCACTGCGGTAATTCGCCGCATCAAAACCGGGCACTTCTGTTTGTTTTAATACACGACGCTCACCGGTATCCATATCCAGTTCAAACAAAGT\tCCCFFFFFHHHHHJJJJJJJJJJJIJJGJJGIIJJIIJIJJJFFDDDDDDDDDDDDDDDDEDDDCDDDDDEEEDDDDDDDDDDDCBDDDB>BDDDDDDEDDCBCCDDDDEDDDDDDBDDBDDDDDDDDDDDDDDDEDDDDDDCDD@DDDC\tNM:i:0\tMD:Z:150\tMC:Z:109M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.1107083\t147\tCP000819.1\t1906756\t60\t109M\t=\t1906108\t-757\tTTTTTCTTCGTAAATCAGTTTATCTTGCGATGCTGGCGTACCGATGGCGTGACGCCAGACCTGATAAGGCAGCAGCGTCACCGGATGCTTGCGAACATAGTAGAAAGTC\t?CCA92BBBCC@@>CDC@AA:(9<<<<<C>3?9=DBB@<@DB>=ACB<B<HHEIGACIJJIHEDJIIIIGCG@?F?HGBBJJJIGIJIJIJIJJIEGHHHFFFFFFCCC\tNM:i:0\tMD:Z:109\tMC:Z:150M\tAS:i:109\tXS:i:0\n",
      "SRR2589044.1107084\t83\tCP000819.1\t1146434\t60\t144M\t=\t1145699\t-879\tTGCGGTAACCACCACCGATCCGACAAATACCGGCGTCAGTACCACCAGCAGTAGTTCGCTCACGGGCAGCAACGCCGCAGATTTACAAAGCAGTTTTCTGACTTTGCTGGTGGCGCAGCTGAAAAACCAGGACCCGACCAATCC\t5&<@CD@9<<BB?<BBB@BBDCDC@CDBDDBDDDDCDDDDB@C?ADCDDCDDBDDDDDDDDB>B?DDDDDDDDDDDDDEDD>FHHHFHFFGGJIHFCJJJJJJJHIJJJJIJJIJJIIIJJIJJJHIJJIIHGHHHFFFDD@C@\tNM:i:0\tMD:Z:144\tMC:Z:59M\tAS:i:144\tXS:i:0\n",
      "SRR2589044.1107084\t163\tCP000819.1\t1145699\t60\t59M\t=\t1146434\t879\tGTTATCAGGCGCGCGATATCGATTTTGCCAGTGAACTTAAAAAAGTCATACAACGTGGA\t??@DDDDBDFHHDEGE1DCCB?;BDGE>9<D<*?FHG=C=CEGE'/=C3=EC=A6?;A=\tNM:i:0\tMD:Z:59\tMC:Z:144M\tAS:i:59\tXS:i:0\n",
      "SRR2589044.1107085\t99\tCP000819.1\t2079336\t60\t125M\t=\t2080020\t752\tGGTGAATGATCTGCTTGAGCAGCTGGAAAATAGCCCTCTACCACCACGGTTATTACATCTGATTATTCCGGCTGAAGCGATTTTAGATCACGCAGAAAGCGTGCAAAAACTGCGGCTGGCGGGAT\tB@@FFFFDHHHHGIJHIJIJIJJJJJIJJJIJJIIIIIIJJJIJJJJJJJJIJIJJJJJJJJJJJJJJIJGFHFFFEEDDDDDDEDDDDDDDDDDDDDCCD@BDDACDDDDDDDBDDBBDDD>.9\tNM:i:0\tMD:Z:125\tMC:Z:68M\tAS:i:125\tXS:i:0\n",
      "SRR2589044.1107085\t147\tCP000819.1\t2080020\t60\t68M\t=\t2079336\t-752\tGGCAGGAAAACATCTTTCGCCTGCCAGCCACGCAAAGCAAAATAATTCGCCGTCCAGCTCCCGATACC\tB>:(@ADDCCA@>8B?8>8@;ACBBBGHHHGGCIIHGEIIIIIHEHBIIIGIGIIFHFAFDFFDF@@@\tNM:i:0\tMD:Z:68\tMC:Z:125M\tAS:i:68\tXS:i:0\n",
      "SRR2589044.1107086\t83\tCP000819.1\t466237\t60\t150M\t=\t465786\t-601\tAAGTCACCGGCGAATCTGGCGCAGGTCTGGTAAAAGTGACCATCAACGGTGCACACAACTGCCGTCGCGTAGAGATCGACCCGAGCCTGCTGGAAGACGACAAAGAGATGCTGGAAGACCTGGTGGCTGCAGCATTCAACGACGCAGCAC\tC@C?BDBDBB@DDC?DDDDDDDDDBB@ADDDDDDCCCCDDDDDDDDDCCDDDDBDDDDDDDDDDDDDDDEEDDDDDDDDDDDFFFHHHHJJJJJJJJJJJIJJJIIJJJJJIJJJJJJJJJJIGJJIJJIJJIIIGBHHHHHFFFFFCCC\tNM:i:0\tMD:Z:150\tMC:Z:100M\tAS:i:150\tXS:i:0\n",
      "SRR2589044.1107086\t163\tCP000819.1\t465786\t60\t100M\t=\t466237\t601\tCCTGGAAAGAGGAGAGCGACAACGCAGTATGTCTGCATTTGCGCTCCTCTCAGCGGCATTTGAACAACCGCGGTGCACAGCAAAAACTGGCTGAAGCGTT\tCCCDFFFFFHDHF@FBFHIJIJJGIIIIIJGIG<GCGIGJ*BFG@BHEG=DEHDFFDABDDCDDCDDADBDBBBDB>A@CDCDDDD-:>AB328CAC@8?\tNM:i:0\tMD:Z:100\tMC:Z:150M\tAS:i:100\tXS:i:0\n",
      "SRR2589044.1107088\t83\tCP000819.1\t2785058\t60\t99M\t=\t2784592\t-565\tGCACCTCCTACATTTCCCGCCGCCACCAGCGGGAGGGCTTTAGTTTCTGGTGCCCATAGCCAGGTCACAACAAAACCAACCAACAACACTGTCGCCAGG\t:2+BA@@>>>CCA95DDDDDDDDDDDDDDDDDDEDEFFFCFHHHHHJJJJIJJIHJJJJJJJJJIGJJJJJJJJJJJJJIJJJJIHHHHHHFFFFFCCC\tNM:i:0\tMD:Z:99\tMC:Z:122M\tAS:i:99\tXS:i:0\n",
      "SRR2589044.1107088\t163\tCP000819.1\t2784592\t60\t122M\t=\t2785058\t565\tAGAGTGAAACGCTCCACCTGGGTGAAGCAGGGCCAGCCCAGCATTTCCGCCAGCAAAAATGGCGTTTGCCCATTCTGCCCTTCGCTGCTTTTGCAGCCGGTAATTATCAGATCCAGCGGATT\t@@CFDFFFHHFHCHIJJJJJJJGIJJGHGGHJIJIIIJIJCHIIIJJJJIJGIJFHHHFDDFEAADB=>CDCCDC>CADD>CC8<05<4>C(:>:@@BBD9<<4>:4@(:>@?CA:C<B99@\tNM:i:1\tMD:Z:91G30\tMC:Z:99M\tAS:i:117\tXS:i:0\n",
      "SRR2589044.1107089\t99\tCP000819.1\t1521593\t60\t44M\t=\t1522446\t952\tGGTATGGCTGGAACAACGGTAGCGGCATTGGGCTTTGCCCCGAA\t@@@DDDDDDFF?F;ABB:GBE)?CGA?:D<FGF>BF<@FFI/;A\tNM:i:0\tMD:Z:44\tMC:Z:99M\tAS:i:44\tXS:i:0\n",
      "SRR2589044.1107089\t147\tCP000819.1\t1522446\t60\t99M\t=\t1521593\t-952\tCCGAATACGTTAAACATTACACCAACGCCAGCCTGCTGGTGCGTGATGATTTTGCTTTCGAAGATGGCCTGTTCAGCGGTTATGACGCTGAAAAACGCC\t?9:?AB??@DCEC@C>A@8@<B@@;DBBB?BA;3;;CCB/E?=)GGHGGHH@CGACHF?IFBD38IJHEIGIHF?HGJJJIIJIJIHHDHHFFFFFCCC\tNM:i:0\tMD:Z:99\tMC:Z:44M\tAS:i:99\tXS:i:0\n",
      "SRR2589044.1107090\t99\tCP000819.1\t3905804\t0\t150M\t=\t3905943\t263\tGTCTGGAAAGGCGCGCGATACAGGGTGACAGCCCCGTACACAAAAATGCACATATTGTGAGCTCGATGAGTAGGGCGGGACACGTGGTATCCTGTCTGAATATGGGGGGACCATCCTCCAAGGCTAAATACTCCTGACTGACCGATAGTG\tBCBFFFFFHHHHHJJIIIDIIJJJJFHJJJIIJJJJJIJJJHHHHFFFFFEEEEEEEEDDDDDDBDDEDDDDDDDCDDDDDDDDABBBDDDDEDDDDDDDDEDEDDDDDBDDDDDDDDDDDD@BDDCDCECDDDDDDDDDDDDB@B9@DE\tNM:i:0\tMD:Z:150\tMC:Z:124M\tAS:i:150\tXS:i:150\n",
      "SRR2589044.1107090\t147\tCP000819.1\t3905943\t0\t124M\t=\t3905804\t-263\tGACCGATAGTGAACCAGTACCGGGAGGGAAAGGCGAAAAGAACCCCGGCGAGGGGAGTGAAAAAGAACCTGAAACCGTGTACGTACAAGCAGTGGGGGCCTCTTTTATGGGGTGACTGCGTACC\t9<92>:4:>>ADDCDC>5@@<2(CCDDCDB>@99CDDDC?BBDDDDD@=@;BDCEEECDFFFFFFHHHHHFIIIHHGAAIGIIHFAIIIIGGBGG?)IGGGCIIIIIIIIIHAGHGFFFFF@C@\tNM:i:2\tMD:Z:22T73A27\tMC:Z:150M\tAS:i:114\tXS:i:114\tXA:Z:CP000819.1,-4190062,124M,2;CP000819.1,+2650113,124M,2;CP000819.1,-4016333,124M,2;CP000819.1,-229029,124M,2;\n"
     ]
    }
   ],
   "source": [
    "!tail -n 15 SRR2589044.aligned.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "bf1f5c10-156b-4c74-b187-de6107020aeb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results/sam'"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8257d1d1-e096-4800-810e-82244a185ac3",
   "metadata": {},
   "source": [
    "## Install sam tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "7b4b7796-f866-48e6-acf2-295b91caa8af",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.7.4\n",
      "  latest version: 23.9.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.9.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/sdogan/anaconda3/envs/NGS_DNAseq_analysis\n",
      "\n",
      "  added / updated specs:\n",
      "    - samtools\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    c-ares-1.20.1              |       hd590300_0         112 KB  conda-forge\n",
      "    htslib-1.18                |       h81da01d_0         2.4 MB  bioconda\n",
      "    libcurl-8.4.0              |       hca28451_0         377 KB  conda-forge\n",
      "    libev-4.33                 |       h516909a_1         104 KB  conda-forge\n",
      "    libnghttp2-1.52.0          |       h61bc06f_0         608 KB  conda-forge\n",
      "    libssh2-1.11.0             |       h0841786_0         265 KB  conda-forge\n",
      "    samtools-1.18              |       h50ea8bc_1         455 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         4.3 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  c-ares             conda-forge/linux-64::c-ares-1.20.1-hd590300_0 \n",
      "  htslib             bioconda/linux-64::htslib-1.18-h81da01d_0 \n",
      "  libcurl            conda-forge/linux-64::libcurl-8.4.0-hca28451_0 \n",
      "  libev              conda-forge/linux-64::libev-4.33-h516909a_1 \n",
      "  libnghttp2         conda-forge/linux-64::libnghttp2-1.52.0-h61bc06f_0 \n",
      "  libssh2            conda-forge/linux-64::libssh2-1.11.0-h0841786_0 \n",
      "  samtools           bioconda/linux-64::samtools-1.18-h50ea8bc_1 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "libssh2-1.11.0       | 265 KB    |                                       |   0% \n",
      "libnghttp2-1.52.0    | 608 KB    |                                       |   0% \u001b[A\n",
      "\n",
      "libev-4.33           | 104 KB    |                                       |   0% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libcurl-8.4.0        | 377 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.18        | 455 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "c-ares-1.20.1        | 112 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.18          | 2.4 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libcurl-8.4.0        | 377 KB    | #5                                    |   4% \u001b[A\u001b[A\u001b[A\n",
      "libnghttp2-1.52.0    | 608 KB    | 9                                     |   3% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.18        | 455 KB    | #3                                    |   4% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "libssh2-1.11.0       | 265 KB    | ##2                                   |   6% \u001b[A\u001b[A\n",
      "\n",
      "libev-4.33           | 104 KB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "libnghttp2-1.52.0    | 608 KB    | ##############################1       |  82% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "c-ares-1.20.1        | 112 KB    | #####2                                |  14% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libcurl-8.4.0        | 377 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libcurl-8.4.0        | 377 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.18          | 2.4 MB    | 2                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "libnghttp2-1.52.0    | 608 KB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.18          | 2.4 MB    | ####################2                 |  55% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.18        | 455 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "c-ares-1.20.1        | 112 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.18        | 455 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libssh2-1.11.0       | 265 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.18          | 2.4 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install samtools"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb7a1625-5189-40a1-be74-ce80e6e13ab5",
   "metadata": {},
   "source": [
    "## Sorting sam files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "d62d66cf-5c1a-4b40-a19e-962119805d03",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!samtools sort SRR2589044.aligned.sam -o SRR2589044.aligned.sorted.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "6dee8d91-6bf2-4348-8920-6aceca9d5b1a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044.aligned.sam\u001b[0m*  \u001b[01;32mSRR2589044.aligned.sorted.sam\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "22300b8a-44f4-4299-801a-60bb76c5a1c5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR2589044.909312\t77\t*\t0\t0\t*\t*\t0\t0\tACCCAAGGTCGGCCGGTACGGATCCGCGTCGACAACGACCTTCCGCAGGCTACGTCGGTGCACTGGCACGGCATCCGCTTGAAAAACGACGCGGATGGGGTGCCCGGCTTCACGCAAGATCCGATCGCGCCCGGCGGGTCGCATCTCTAC\tCCCFFFFFHHHHHJJJIJIJJJGIJJJJGIIIJHHFFFDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD5?CDDDDDDDDDDDDDDDCDDDD@DDDDDDDDBBDDB@BBB@BDDCEED\tAS:i:0\tXS:i:0\n",
      "SRR2589044.909312\t141\t*\t0\t0\t*\t*\t0\t0\tTCGTAGGACAGCTCCTCTCGGCGCGCGTCGACGATGAGCGCGCCATAGAGGCCGCGGTCCAGCTGGGTGCCGACGTGGGTGTGGAAGAAGTAGGTCCCGGCGTCCGGGGGTGTGAATTCGTAGAGATGCGACCCGCCGGGCGCGATCGGA\tBCCFFFFFHHHHHJJJJGIIJJIJJJIHJIIJHHFFFFEDDDDDDDDDDDDDDDDDDDDDDDDDDDDABDDDDDD>B?B5@>BDCDDBDDACCDDDCDBDDDDDDDDDDD995@CEEDDDCCDDC@CDDDDDDDDDDDBDBDDD>@BD<?\tAS:i:0\tXS:i:0\n",
      "SRR2589044.933150\t77\t*\t0\t0\t*\t*\t0\t0\tTCAGCATATTGAACAACTCATCCAATTTG\t@@?DDDDDHH>DD>H;EAHEECAE@CHEF\tAS:i:0\tXS:i:0\n",
      "SRR2589044.933150\t141\t*\t0\t0\t*\t*\t0\t0\tGCCTCAATTTAACCGTCGTGGCGGAAAGCG\t<?@DDD?ADHF>CFHBFBDAH>EH6:@8*7\tAS:i:0\tXS:i:0\n",
      "SRR2589044.966182\t77\t*\t0\t0\t*\t*\t0\t0\tGTCCCGGACAGCCCCGACCTGGGCCCGCTGGTCAAGACGCTGCTGGACACCATCCCGGCCCCGGTCTACGACGCCGAGGAGCCCCTGCGCGCGCAGGTCACCAACCTCGACGCCTCGCCCTTCCTCGGCCGCCTCGCGCTGCTGCGCATC\tCCCFFFFFHHHHHJJJJJJJJJJJJJJJIJJJJJJJJJHHHFFFEDEEEDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDBDDDDDDDDDDDDDCCABDDDDDDDDDDDDDDDDDDDDDBDDDDBDBDDDDDDBDDDDDBDDD\tAS:i:0\tXS:i:0\n",
      "SRR2589044.966182\t141\t*\t0\t0\t*\t*\t0\t0\tGGCGCGGGTCGTCGGGGTCGGCCAGGGTGTCGCCGATGCCGATGT\tCCCFFFFFFHHHHJJJJIJIJJJHHHF??CEBDDDDDBDDDDDDD\tAS:i:0\tXS:i:0\n",
      "SRR2589044.1026760\t77\t*\t0\t0\t*\t*\t0\t0\tGCAGAAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAA\t@@@DDDD?+CFDFFEF@FICFGF<CHEBFC>@FFFFBFI\tAS:i:0\tXS:i:0\n",
      "SRR2589044.1026760\t141\t*\t0\t0\t*\t*\t0\t0\tTGTGGAACCACAGCGTTTCCGTGGTGG\t;?@DBDDD;+A?FE)<AECH=E;)88?\tAS:i:0\tXS:i:0\n",
      "SRR2589044.1086438\t77\t*\t0\t0\t*\t*\t0\t0\tGGGCTGAGCCGGGGGCGGGCCATGCGGCCCGTCCGAGGGTGCCCGGTCCAGTGGTTGCGATCGTCGTTTCCGGCTGGTCTACTGCGCTGAGTCTGCCGCGCGTGTGCGGCATCGGTTCCCGTCGTATCCCGAGGAGTTCGCGTGTCCCGT\t@@CFFFDFHHFFDGGIGBBBDBDDCDDDDBBDBDDDD6@?<@CDDDBBDDD@ACCDDDDBBDDCDD@BDDDDDD9<@?C>@CDDCBDBBBDDDC@CDBB.>59@<BC>@B95?@DBBBACBDDD@BC?CC@B5>389CC?>BDDBB>CB8\tAS:i:0\tXS:i:0\n",
      "SRR2589044.1086438\t141\t*\t0\t0\t*\t*\t0\t0\tCGCCACGACACCCGGGATCAGCACGGCGACGACGGGGACCCAGGGCTTCCTGGGGTCAGGCATCGATGCCACCAGCGGCGGCGAGCTTGGG\t@@@DDFDFFHHHHIIIHBEHGIJJIIIDBHA?<3>@BDDDD?AB@BBBCCCCAB65:?ADDBCCBCD<@DCCBBB<<BDD9>@3595>??B\tAS:i:0\tXS:i:0\n"
     ]
    }
   ],
   "source": [
    "!tail SRR2589044.aligned.sorted.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd4f5385-c9ff-45cf-898f-4cc8c5c75f6d",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Convert sam files to bam files # -S > sam file -b > bam file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "88aa46e5-e8d6-40a5-b339-3c59dcb7eeb5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!samtools view -S -b SRR2589044.aligned.sam > ../bam/SRR2589044.aligned.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "dc063f1f-1ac1-4387-a170-ed7a6eac57b0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "6fff27ae-4ae3-4984-a853-d863e3a8769c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[34;42mbam\u001b[0m/  \u001b[34;42mbcf\u001b[0m/  \u001b[34;42mfastqc\u001b[0m/  \u001b[34;42msam\u001b[0m/  \u001b[34;42mtrimmomatic\u001b[0m/  \u001b[34;42mvcf\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e744e5e-9866-4b8f-95e6-ae1e983a2498",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Sorting bam files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "5e77fb6d-4156-43da-8dff-5eb642e27ac5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!samtools sort bam/SRR2589044.aligned.bam -o bam/SRR2589044.aligned.sorted.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d3b0f90-2a81-4ece-ae4b-5503f8805730",
   "metadata": {},
   "source": [
    "## Indexing the sorted aligned files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "864b3fd8-2a0e-4005-a0bd-c392477fafcb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!samtools index bam/SRR2589044.aligned.sorted.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "895d9fb8-c2a7-46ec-a8bf-980a673d1f4d",
   "metadata": {},
   "source": [
    "## Alignment stats # Generate Alignmnet Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "57fb67c8-2d62-4800-a0e4-0f752fe3f2e8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1789959 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "1770440 + 0 primary\n",
      "0 + 0 secondary\n",
      "19519 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "1788837 + 0 mapped (99.94% : N/A)\n",
      "1769318 + 0 primary mapped (99.94% : N/A)\n",
      "1770440 + 0 paired in sequencing\n",
      "885220 + 0 read1\n",
      "885220 + 0 read2\n",
      "1696562 + 0 properly paired (95.83% : N/A)\n",
      "1768266 + 0 with itself and mate mapped\n",
      "1052 + 0 singletons (0.06% : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "!samtools flagstat bam/SRR2589044.aligned.sorted.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f488cee8-743c-4d72-b807-559d7a3cf66c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Calling the variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "7622ecb2-f0f3-4e56-a302-277204576e32",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e363a2f6-7396-4154-b38c-56a73023e5a1",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Install bcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "4ead7905-c531-4d02-b9fc-35c53ce70d53",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.7.4\n",
      "  latest version: 23.9.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.9.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/sdogan/anaconda3/envs/NGS_DNAseq_analysis\n",
      "\n",
      "  added / updated specs:\n",
      "    - bcftools\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    bcftools-1.18              |       h8b25389_0         939 KB  bioconda\n",
      "    gsl-2.7                    |       he838d99_0         3.2 MB  conda-forge\n",
      "    libblas-3.9.0              |19_linux64_openblas          14 KB  conda-forge\n",
      "    libcblas-3.9.0             |19_linux64_openblas          14 KB  conda-forge\n",
      "    libgfortran-ng-13.2.0      |       h69a702a_2          23 KB  conda-forge\n",
      "    libgfortran5-13.2.0        |       ha4646dd_2         1.4 MB  conda-forge\n",
      "    libopenblas-0.3.24         |pthreads_h413a1c8_0         5.2 MB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        10.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  bcftools           bioconda/linux-64::bcftools-1.18-h8b25389_0 \n",
      "  gsl                conda-forge/linux-64::gsl-2.7-he838d99_0 \n",
      "  libblas            conda-forge/linux-64::libblas-3.9.0-19_linux64_openblas \n",
      "  libcblas           conda-forge/linux-64::libcblas-3.9.0-19_linux64_openblas \n",
      "  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-13.2.0-h69a702a_2 \n",
      "  libgfortran5       conda-forge/linux-64::libgfortran5-13.2.0-ha4646dd_2 \n",
      "  libopenblas        conda-forge/linux-64::libopenblas-0.3.24-pthreads_h413a1c8_0 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "libopenblas-0.3.24   | 5.2 MB    |                                       |   0% \n",
      "libgfortran-ng-13.2. | 23 KB     |                                       |   0% \u001b[A\n",
      "\n",
      "gsl-2.7              | 3.2 MB    |                                       |   0% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libcblas-3.9.0       | 14 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "libblas-3.9.0        | 14 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libgfortran5-13.2.0  | 1.4 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bcftools-1.18        | 939 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "libblas-3.9.0        | 14 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | 1                                     |   0% \u001b[A\u001b[A\n",
      "libgfortran-ng-13.2. | 23 KB     | #########################5            |  69% \u001b[A\n",
      "\n",
      "\n",
      "libcblas-3.9.0       | 14 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libgfortran5-13.2.0  | 1.4 MB    | 4                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "libblas-3.9.0        | 14 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | #####5                                |  15% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | ###4                                  |   9% \u001b[A\u001b[A\u001b[A\n",
      "libgfortran-ng-13.2. | 23 KB     | ##################################### | 100% \u001b[A\n",
      "libgfortran-ng-13.2. | 23 KB     | ##################################### | 100% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bcftools-1.18        | 939 KB    | 6                                     |   2% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libgfortran5-13.2.0  | 1.4 MB    | ######3                               |  17% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | #####6                                |  15% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bcftools-1.18        | 939 KB    | ######9                               |  19% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | ###############4                      |  42% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | ########3                             |  23% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bcftools-1.18        | 939 KB    | ############6                         |  34% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | ######################4               |  61% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | ##########5                           |  29% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bcftools-1.18        | 939 KB    | #################6                    |  48% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | #############4                        |  36% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libgfortran5-13.2.0  | 1.4 MB    | ###############################5      |  85% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bcftools-1.18        | 939 KB    | ######################                |  60% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | #################3                    |  47% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | ######################4               |  61% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | ###############################8      |  86% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "libopenblas-0.3.24   | 5.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install bcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "98669f2e-049a-4896-8920-6ea77f29186e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "channels:\n",
      "  - r\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n"
     ]
    }
   ],
   "source": [
    "!conda config --show channels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "72a741de-a881-43f8-863e-d319722f2b3b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Calculate the read coverage of positions in the genome\n",
    "# -O b means output file in bcf format\n",
    "# -o directs output to a specific folder\n",
    "# -f tags path to reference genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "09f2d046-2f36-437c-8ef7-6a17e11e1a29",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n"
     ]
    }
   ],
   "source": [
    "!bcftools mpileup -O b -f reference/ecoli_rel606.fasta \\\n",
    "results/bam/SRR2589044.aligned.sorted.bam \\\n",
    "-o results/bcf/SRR2589044_raw.bcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "853e2c17-1256-4905-b797-f40e7780e2fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calling the variants\n",
    "# ploidy - 1 for bacteria\n",
    "# -m allows for multiallelic and rare-variant calling\n",
    "# -v tells the program to output variant sites only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "193a43bc-bf50-4a99-aeb6-1f7685f0daed",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools call --ploidy 1 -m -v results/bcf/SRR2589044_raw.bcf \\\n",
    "-o results/vcf/SRR2589044_variants.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1faee29-4504-41e5-a2c9-5ed4eb94bf80",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Checking the raw variant file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "cd5189a4-38bc-453a-a47b-63e1ad05b075",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "##bcftoolsVersion=1.18+htslib-1.18\n",
      "##bcftoolsCommand=mpileup -O b -f reference/ecoli_rel606.fasta -o results/bcf/SRR2589044_raw.bcf results/bam/SRR2589044.aligned.sorted.bam\n",
      "##reference=file://reference/ecoli_rel606.fasta\n",
      "##contig=<ID=CP000819.1,length=4629812>\n",
      "##ALT=<ID=*,Description=\"Represents allele(s) other than observed.\">\n",
      "##INFO=<ID=INDEL,Number=0,Type=Flag,Description=\"Indicates that the variant is an INDEL.\">\n",
      "##INFO=<ID=IDV,Number=1,Type=Integer,Description=\"Maximum number of raw reads supporting an indel\">\n",
      "##INFO=<ID=IMF,Number=1,Type=Float,Description=\"Maximum fraction of raw reads supporting an indel\">\n"
     ]
    }
   ],
   "source": [
    "!head results/vcf/SRR2589044_variants.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9dc07235-3928-477f-840b-957d5328e052",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "b1eda024-15da-4dc2-bac2-a597895be8e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "!vcfutils.pl varFilter results/vcf/SRR2589044_variants.vcf > results/vcf/SRR2589044_final_variants.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "c3da913c-a180-4945-b181-291c84e7da18",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "##bcftoolsVersion=1.18+htslib-1.18\n",
      "##bcftoolsCommand=mpileup -O b -f reference/ecoli_rel606.fasta -o results/bcf/SRR2589044_raw.bcf results/bam/SRR2589044.aligned.sorted.bam\n",
      "##reference=file://reference/ecoli_rel606.fasta\n",
      "##contig=<ID=CP000819.1,length=4629812>\n",
      "##ALT=<ID=*,Description=\"Represents allele(s) other than observed.\">\n",
      "##INFO=<ID=INDEL,Number=0,Type=Flag,Description=\"Indicates that the variant is an INDEL.\">\n",
      "##INFO=<ID=IDV,Number=1,Type=Integer,Description=\"Maximum number of raw reads supporting an indel\">\n",
      "##INFO=<ID=IMF,Number=1,Type=Float,Description=\"Maximum fraction of raw reads supporting an indel\">\n"
     ]
    }
   ],
   "source": [
    "!head results/vcf/SRR2589044_final_variants.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "7b2768e6-80f5-4dd3-8a68-7cb50652b576",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CP000819.1\t762231\t.\tT\tG\t38.415\t.\tDP=2;VDB=0.02;SGB=-0.453602;MQSBZ=-1;MQ0F=0;AC=1;AN=1;DP4=0,0,1,1;MQ=52\tGT:PL\t1:68,0\n",
      "CP000819.1\t1331794\t.\tC\tA\t225.417\t.\tDP=34;VDB=0.126039;SGB=-0.693127;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,16,17;MQ=60\tGT:PL\t1:255,0\n",
      "CP000819.1\t1733343\t.\tG\tA\t225.417\t.\tDP=66;VDB=0.997802;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,33,28;MQ=60\tGT:PL\t1:255,0\n",
      "CP000819.1\t2103887\t.\tACAGCCAGCCAGCCAGCCAGCCAGCCAGCCAG\tACAGCCAGCCAGCCAGCCAGCCAGCCAGCCAGCCAG\t207.185\t.\tINDEL;IDV=27;IMF=0.72973;DP=37;VDB=0.011574;SGB=-0.693021;RPBZ=-2.77162;MQBZ=0;MQSBZ=0;BQBZ=-0.998006;SCBZ=-2.35584;MQ0F=0;AC=1;AN=1;DP4=2,6,9,18;MQ=60\tGT:PL\t1:255,21\n",
      "CP000819.1\t3444175\t.\tG\tT\t225.417\t.\tDP=44;VDB=0.0199106;SGB=-0.693145;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,20,21;MQ=60\tGT:PL\t1:255,0\n",
      "CP000819.1\t3481820\t.\tA\tG\t225.417\t.\tDP=51;VDB=0.959152;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,19,31;MQ=60\tGT:PL\t1:255,0\n",
      "CP000819.1\t3893550\t.\tAG\tAGG\t228.357\t.\tINDEL;IDV=42;IMF=0.857143;DP=49;VDB=0.843309;SGB=-0.693146;RPBZ=0.600613;MQBZ=3.17582;MQSBZ=1.18597;BQBZ=-1.71415;SCBZ=-5.05621;MQ0F=0.0204082;AC=1;AN=1;DP4=3,2,26,16;MQ=54\tGT:PL\t1:255,0\n",
      "CP000819.1\t3901455\t.\tA\tAC\t107.415\t.\tINDEL;IDV=5;IMF=1;DP=5;VDB=0.00215942;SGB=-0.590765;MQ0F=0;AC=1;AN=1;DP4=0,0,5,0;MQ=60\tGT:PL\t1:137,0\n",
      "CP000819.1\t4100183\t.\tA\tG\t225.417\t.\tDP=48;VDB=0.996765;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,27,21;MQ=60\tGT:PL\t1:255,0\n",
      "CP000819.1\t4431393\t.\tTGG\tT\t228.417\t.\tINDEL;IDV=56;IMF=0.965517;DP=58;VDB=0.0260664;SGB=-0.693147;RPBZ=-2.36673;MQBZ=0;MQSBZ=0;BQBZ=-1.72278;SCBZ=-5.2915;MQ0F=0;AC=1;AN=1;DP4=2,0,27,29;MQ=60\tGT:PL\t1:255,0\n"
     ]
    }
   ],
   "source": [
    "!tail results/vcf/SRR2589044_final_variants.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92f83761-10e3-42aa-a004-be175f686b2b",
   "metadata": {},
   "source": [
    "## The Integrative Genomics Viewer (IGV) is a high-performance, easy-to-use, interactive tool for the visual exploration of genomic data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4101e42b-34bd-4cb8-a2af-12e829c1ac5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download IGV and install into your computer\n",
    "# https://igv.org/doc/desktop/#DownloadPage/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f4d6312-dc5f-4d6d-9fa7-3369b5c768e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Open IGV software and Upload your reference genome reference > ecoli_rel606.fasta\n",
    "# Genome > Load Genome from File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "007cef9e-3955-4bfe-bcda-df889775ae1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load alignment file >.bam file > index file bam.bai should be there also\n",
    "# Load from file > results > bam > SRR2589044.aligned.sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "85af2a21-1796-49fb-800f-e3f6a1f6652f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/sdogan/Desktop/NGS_DNAseq_analysis/results/vcf\n"
     ]
    }
   ],
   "source": [
    "cd results/vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "a05198b7-5a8b-4aa3-a6ca-d036321d2dc1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mSRR2589044_final_variants.vcf\u001b[0m*  \u001b[01;32mSRR2589044_variants.vcf\u001b[0m*\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "3aa363d0-ab15-439d-9217-ac817e6187da",
   "metadata": {},
   "outputs": [],
   "source": [
    "## open your variants file and check the positions "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "31c9d0fb-10e6-4d53-a29c-be0ba458bbae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "##fileformat=VCFv4.2\n",
       "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
       "##bcftoolsVersion=1.18+htslib-1.18\n",
       "##bcftoolsCommand=mpileup -O b -f reference/ecoli_rel606.fasta -o results/bcf/SRR2589044_raw.bcf results/bam/SRR2589044.aligned.sorted.bam\n",
       "##reference=file://reference/ecoli_rel606.fasta\n",
       "##contig=<ID=CP000819.1,length=4629812>\n",
       "##ALT=<ID=*,Description=\"Represents allele(s) other than observed.\">\n",
       "##INFO=<ID=INDEL,Number=0,Type=Flag,Description=\"Indicates that the variant is an INDEL.\">\n",
       "##INFO=<ID=IDV,Number=1,Type=Integer,Description=\"Maximum number of raw reads supporting an indel\">\n",
       "##INFO=<ID=IMF,Number=1,Type=Float,Description=\"Maximum fraction of raw reads supporting an indel\">\n",
       "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Raw read depth\">\n",
       "##INFO=<ID=VDB,Number=1,Type=Float,Description=\"Variant Distance Bias for filtering splice-site artefacts in RNA-seq data (bigger is better)\",Version=\"3\">\n",
       "##INFO=<ID=RPBZ,Number=1,Type=Float,Description=\"Mann-Whitney U-z test of Read Position Bias (closer to 0 is better)\">\n",
       "##INFO=<ID=MQBZ,Number=1,Type=Float,Description=\"Mann-Whitney U-z test of Mapping Quality Bias (closer to 0 is better)\">\n",
       "##INFO=<ID=BQBZ,Number=1,Type=Float,Description=\"Mann-Whitney U-z test of Base Quality Bias (closer to 0 is better)\">\n",
       "##INFO=<ID=MQSBZ,Number=1,Type=Float,Description=\"Mann-Whitney U-z test of Mapping Quality vs Strand Bias (closer to 0 is better)\">\n",
       "##INFO=<ID=SCBZ,Number=1,Type=Float,Description=\"Mann-Whitney U-z test of Soft-Clip Length Bias (closer to 0 is better)\">\n",
       "##INFO=<ID=SGB,Number=1,Type=Float,Description=\"Segregation based metric, http://samtools.github.io/bcftools/rd-SegBias.pdf\">\n",
       "##INFO=<ID=MQ0F,Number=1,Type=Float,Description=\"Fraction of MQ0 reads (smaller is better)\">\n",
       "##FORMAT=<ID=PL,Number=G,Type=Integer,Description=\"List of Phred-scaled genotype likelihoods\">\n",
       "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
       "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes for each ALT allele, in the same order as listed\">\n",
       "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles in called genotypes\">\n",
       "##INFO=<ID=DP4,Number=4,Type=Integer,Description=\"Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases\">\n",
       "##INFO=<ID=MQ,Number=1,Type=Integer,Description=\"Average mapping quality\">\n",
       "##bcftools_callVersion=1.18+htslib-1.18\n",
       "##bcftools_callCommand=call --ploidy 1 -m -v -o results/vcf/SRR2589044_variants.vcf results/bcf/SRR2589044_raw.bcf; Date=Sat Oct 21 21:35:36 2023\n",
       "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tresults/bam/SRR2589044.aligned.sorted.bam\n",
       "CP000819.1\t648692\t.\tC\tT\t225.417\t.\tDP=44;VDB=0.1874;SGB=-0.693146;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,32,11;MQ=60\tGT:PL\t1:255,0\n",
       "CP000819.1\t762231\t.\tT\tG\t38.415\t.\tDP=2;VDB=0.02;SGB=-0.453602;MQSBZ=-1;MQ0F=0;AC=1;AN=1;DP4=0,0,1,1;MQ=52\tGT:PL\t1:68,0\n",
       "CP000819.1\t1331794\t.\tC\tA\t225.417\t.\tDP=34;VDB=0.126039;SGB=-0.693127;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,16,17;MQ=60\tGT:PL\t1:255,0\n",
       "CP000819.1\t1733343\t.\tG\tA\t225.417\t.\tDP=66;VDB=0.997802;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,33,28;MQ=60\tGT:PL\t1:255,0\n",
       "CP000819.1\t2103887\t.\tACAGCCAGCCAGCCAGCCAGCCAGCCAGCCAG\tACAGCCAGCCAGCCAGCCAGCCAGCCAGCCAGCCAG\t207.185\t.\tINDEL;IDV=27;IMF=0.72973;DP=37;VDB=0.011574;SGB=-0.693021;RPBZ=-2.77162;MQBZ=0;MQSBZ=0;BQBZ=-0.998006;SCBZ=-2.35584;MQ0F=0;AC=1;AN=1;DP4=2,6,9,18;MQ=60\tGT:PL\t1:255,21\n",
       "CP000819.1\t3444175\t.\tG\tT\t225.417\t.\tDP=44;VDB=0.0199106;SGB=-0.693145;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,20,21;MQ=60\tGT:PL\t1:255,0\n",
       "CP000819.1\t3481820\t.\tA\tG\t225.417\t.\tDP=51;VDB=0.959152;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,19,31;MQ=60\tGT:PL\t1:255,0\n",
       "CP000819.1\t3893550\t.\tAG\tAGG\t228.357\t.\tINDEL;IDV=42;IMF=0.857143;DP=49;VDB=0.843309;SGB=-0.693146;RPBZ=0.600613;MQBZ=3.17582;MQSBZ=1.18597;BQBZ=-1.71415;SCBZ=-5.05621;MQ0F=0.0204082;AC=1;AN=1;DP4=3,2,26,16;MQ=54\tGT:PL\t1:255,0\n",
       "CP000819.1\t3901455\t.\tA\tAC\t107.415\t.\tINDEL;IDV=5;IMF=1;DP=5;VDB=0.00215942;SGB=-0.590765;MQ0F=0;AC=1;AN=1;DP4=0,0,5,0;MQ=60\tGT:PL\t1:137,0\n",
       "CP000819.1\t4100183\t.\tA\tG\t225.417\t.\tDP=48;VDB=0.996765;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,27,21;MQ=60\tGT:PL\t1:255,0\n",
       "CP000819.1\t4431393\t.\tTGG\tT\t228.417\t.\tINDEL;IDV=56;IMF=0.965517;DP=58;VDB=0.0260664;SGB=-0.693147;RPBZ=-2.36673;MQBZ=0;MQSBZ=0;BQBZ=-1.72278;SCBZ=-5.2915;MQ0F=0;AC=1;AN=1;DP4=2,0,27,29;MQ=60\tGT:PL\t1:255,0\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "more SRR2589044_final_variants.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90ee27f3-1459-4284-a37e-b5185e0ee51d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# CP000819.1\t648692\t.\tC\tT\t225.417\t.\tDP=44;VDB=0.1874;SGB=-0.693146;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,32,11;MQ=60\tGT:PL\t1:255,0\n",
    "# in IGV go to 648692 position > define interval > CP000819.1:648600-648700\n",
    "# paste your interval CP000819.1:648600-648700 into IGV > Go to see you region of interest > your variant position\n",
    "# left click on each read shows the new window containing read details"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
